--- a/configure.ac
+++ b/configure.ac
@@ -12,12 +12,13 @@
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
 AC_CONFIG_HEADERS([config.h])
 
-SPELLER_LIB=-lenchant
-      
-AC_SUBST(SPELLER_LIB)
 GTKSPELL_PACKAGES=gtk+-2.0
 AC_SUBST(GTKSPELL_PACKAGES)
-PKG_CHECK_MODULES(GTKSPELL, $GTKSPELL_PACKAGES enchant >= 0.4.0 )
+PKG_CHECK_MODULES([GTKSPELL], $GTKSPELL_PACKAGES enchant-2 >= 2.0.0,
+   [HAVE_ENCHANT2="yes";AC_DEFINE([HAVE_ENCHANT2], [1], [Set if enchant-2 library is available.])],
+   [PKG_CHECK_MODULES([GTKSPELL], $GTKSPELL_PACKAGES enchant >= 0.4.0)
+])
+AM_CONDITIONAL(HAVE_ENCHANT2, test "$HAVE_ENCHANT2" = "yes")
 AC_SUBST(GTKSPELL_CFLAGS)
 AC_SUBST(GTKSPELL_LIBS)
 
--- a/gtkspell/Makefile.am
+++ b/gtkspell/Makefile.am
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES=libgtkspell.la
 AM_CFLAGS=@GTKSPELL_CFLAGS@ -DG_LOG_DOMAIN=\"gtkspell\" -DLOCALEDIR=\""$(datadir)/locale"\"
-libgtkspell_la_LIBADD=@GTKSPELL_LIBS@ @SPELLER_LIB@
+libgtkspell_la_LIBADD=@GTKSPELL_LIBS@
 libgtkspell_la_includedir=$(includedir)/gtkspell-2.0/gtkspell
 libgtkspell_la_include_HEADERS=gtkspell.h
 libgtkspell_la_SOURCES=gtkspell.c deprecated.c gtkspell.h
--- a/gtkspell/gtkspell.c
+++ b/gtkspell/gtkspell.c
@@ -277,7 +277,11 @@ add_to_dictionary(GtkWidget *menuitem, GtkSpell *spell) {
 	get_word_extents_from_mark(spell->buffer, &start, &end, spell->mark_click);
 	word = gtk_text_buffer_get_text(spell->buffer, &start, &end, FALSE);
 	
+#ifdef HAVE_ENCHANT2
+	enchant_dict_add( spell->speller, word, strlen(word));
+#else
 	enchant_dict_add_to_pwl( spell->speller, word, strlen(word));
+#endif
 
 	gtkspell_recheck_all(spell);
 
