diff --git a/gtk/src/audiohandler.c b/gtk/src/audiohandler.c
index e922d40f740c53d6258399982a7ece477e9c0f1e..404c37659cfddd437f12eca4a3a844488dfb780a 100644
--- a/gtk/src/audiohandler.c
+++ b/gtk/src/audiohandler.c
@@ -24,7 +24,7 @@
 
 #include <glib/gi18n.h>
 #include "ghbcompat.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "settings.h"
 #include "jobdict.h"
 #include "titledict.h"
diff --git a/gtk/src/callbacks.c b/gtk/src/callbacks.c
index b8c41321ece466a2d579fb1c60b1b4c50da478fe..0a9442c9bc4c42744b0c65e8c77a3041eaef487c 100644
--- a/gtk/src/callbacks.c
+++ b/gtk/src/callbacks.c
@@ -74,7 +74,7 @@
 #include <dbt.h>
 #endif
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "callbacks.h"
 #include "chapters.h"
 #include "queuehandler.h"
diff --git a/gtk/src/callbacks.h b/gtk/src/callbacks.h
index 94f64bb742a3e211cf7672a4fec9f9e61faac09a..d2390e0ff26b719c2718052a9c3bd78b11980e23 100644
--- a/gtk/src/callbacks.h
+++ b/gtk/src/callbacks.h
@@ -30,7 +30,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "values.h"
 #include "settings.h"
 
diff --git a/gtk/src/hb-backend.c b/gtk/src/hb-backend.c
index 260f09034a22e435534b5843f96f60066e885a84..535d60f93f8bfad974dfeed7ddb8d107490e692d 100644
--- a/gtk/src/hb-backend.c
+++ b/gtk/src/hb-backend.c
@@ -25,7 +25,7 @@
 #include <limits.h>
 #include <ctype.h>
 #include <math.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "ghbcompat.h"
 #include <glib/gstdio.h>
 #include <glib/gi18n.h>
@@ -39,7 +39,7 @@
 #include "preview.h"
 #include "presets.h"
 #include "values.h"
-#include "lang.h"
+#include "handbrake/lang.h"
 #include "jansson.h"
 
 typedef struct
diff --git a/gtk/src/hb-backend.h b/gtk/src/hb-backend.h
index 37d21c14a2486399c34ac46b1738e09a551532b5..38b1545a8ed997a4646f5fa0b937ad93e1750963 100644
--- a/gtk/src/hb-backend.h
+++ b/gtk/src/hb-backend.h
@@ -26,8 +26,8 @@
 
 #include "values.h"
 #include "settings.h"
-#include "handbrake.h"
-#include "lang.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/lang.h"
 
 enum
 {
diff --git a/gtk/src/main.c b/gtk/src/main.c
index 7b59bc4608d326d0a70aea28999e4c2f5ae7e8d9..c7c4227243094c7ef3aea22f27d75ae68233228a 100644
--- a/gtk/src/main.c
+++ b/gtk/src/main.c
@@ -47,7 +47,7 @@
 #include <glib/gstdio.h>
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "renderer_button.h"
 #include "hb-backend.h"
 #include "ghb-dvd.h"
diff --git a/gtk/src/presets.c b/gtk/src/presets.c
index f06f4552d96ad21a8ab2150a86f13ab2187a3a5d..8e94cfb6c9e3fb52bdb33240b731d4a10ae23e11 100644
--- a/gtk/src/presets.c
+++ b/gtk/src/presets.c
@@ -31,7 +31,7 @@
 #include <glib/gi18n.h>
 #include <string.h>
 #include "ghbcompat.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "settings.h"
 #include "callbacks.h"
 #include "audiohandler.h"
@@ -40,7 +40,7 @@
 #include "resources.h"
 #include "presets.h"
 #include "values.h"
-#include "lang.h"
+#include "handbrake/lang.h"
 #include "videohandler.h"
 
 #define MAX_NESTED_PRESET 3
diff --git a/gtk/src/presets.h b/gtk/src/presets.h
index 32011e4e2a1268889c1008179ca872f06c8a19f8..aed1d5d56cc317f423911a35067859100b477302 100644
--- a/gtk/src/presets.h
+++ b/gtk/src/presets.h
@@ -24,7 +24,7 @@
 #if !defined(_GHB_PRESETS_H_)
 #define _GHB_PRESETS_H_
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "values.h"
 
 void ghb_presets_load(signal_user_data_t *ud);
diff --git a/gtk/src/preview.c b/gtk/src/preview.c
index 7ae9563550d14d0b8234adc7b3dc34923da1de17..be97edbe8e59e8f10a557a34cee081048a621430 100644
--- a/gtk/src/preview.c
+++ b/gtk/src/preview.c
@@ -43,7 +43,7 @@
 #include "preview.h"
 #include "values.h"
 #include "queuehandler.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #define PREVIEW_STATE_IMAGE 0
 #define PREVIEW_STATE_LIVE 1
diff --git a/gtk/src/queuehandler.c b/gtk/src/queuehandler.c
index efc8292db77c617ebaa15afe27cd3fb1063043a3..ace9064dd6c934dde379ecbf16ce9e7d3576c9b1 100644
--- a/gtk/src/queuehandler.c
+++ b/gtk/src/queuehandler.c
@@ -26,7 +26,7 @@
 #include <glib/gstdio.h>
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "settings.h"
 #include "jobdict.h"
 #include "titledict.h"
diff --git a/gtk/src/subtitlehandler.c b/gtk/src/subtitlehandler.c
index 5e31f684f6c720a4c5595d76a313051be7cd3c32..d7d0c7e9cf420a0b879898696fcee2146cd725f8 100644
--- a/gtk/src/subtitlehandler.c
+++ b/gtk/src/subtitlehandler.c
@@ -24,7 +24,7 @@
 
 #include <glib/gi18n.h>
 #include "ghbcompat.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "settings.h"
 #include "jobdict.h"
 #include "titledict.h"
diff --git a/gtk/src/values.h b/gtk/src/values.h
index fabc2021f09fcf67b96af3cd99885fd8473fbfb7..a6369a7a1baf5f37979eb3a0be0b9203cdb629fa 100644
--- a/gtk/src/values.h
+++ b/gtk/src/values.h
@@ -26,7 +26,7 @@
 
 #include <glib.h>
 #include <glib-object.h>
-#include "hb_dict.h"
+#include "handbrake/hb_dict.h"
 
 #define GHB_DICT    HB_VALUE_TYPE_DICT
 #define GHB_ARRAY   HB_VALUE_TYPE_ARRAY
diff --git a/libhb/audio_remap.c b/libhb/audio_remap.c
index b480a69dd41a6741a1f76e56eb0c80d5b08d7147..e44a9572c5975374c4ff6be3636d8baa9c8b1271 100644
--- a/libhb/audio_remap.c
+++ b/libhb/audio_remap.c
@@ -7,9 +7,9 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "hbffmpeg.h"
-#include "audio_remap.h"
+#include "handbrake/common.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/audio_remap.h"
 
 // source: libavutil/channel_layout.h
 hb_chan_map_t hb_libav_chan_map =
diff --git a/libhb/audio_resample.c b/libhb/audio_resample.c
index d415b574aa786435b892055999188e354a3836a1..3c1bb6f006d020bad6732585f7abc98ae128a3e7 100644
--- a/libhb/audio_resample.c
+++ b/libhb/audio_resample.c
@@ -7,9 +7,9 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "hbffmpeg.h"
-#include "audio_resample.h"
+#include "handbrake/common.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/audio_resample.h"
 
 hb_audio_resample_t* hb_audio_resample_init(enum AVSampleFormat sample_fmt,
                                             int sample_rate,
diff --git a/libhb/avfilter.c b/libhb/avfilter.c
index d7fe4ea04ea63cc41a30e8294c7ad6e74c4650bc..64dab837165be8f4ca92b6b06c2a581317638dbe 100644
--- a/libhb/avfilter.c
+++ b/libhb/avfilter.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "hbavfilter.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/hbavfilter.h"
+#include "handbrake/avfilter_priv.h"
 
 static int  avfilter_init(hb_filter_object_t * filter, hb_filter_init_t * init);
 static int  avfilter_post_init( hb_filter_object_t * filter, hb_job_t * job );
diff --git a/libhb/batch.c b/libhb/batch.c
index ca61d39321e60eea08420bd0743d38bb3e3e239f..d633d83b85c29ceedd72f437911120b27720f567 100644
--- a/libhb/batch.c
+++ b/libhb/batch.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
-#include "handbrake.h"
-#include "lang.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/lang.h"
 
 struct hb_batch_s
 {
diff --git a/libhb/bd.c b/libhb/bd.c
index ba230d98fc5d893885c6b4abd55a9b448ad3cb9d..c5a41c7fc8cf3eff8ee7613eae63673618b68100 100644
--- a/libhb/bd.c
+++ b/libhb/bd.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "lang.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/lang.h"
+#include "handbrake/hbffmpeg.h"
 
 #include "libbluray/bluray.h"
 
diff --git a/libhb/chroma_smooth.c b/libhb/chroma_smooth.c
index 0cbfbeea10d289f6cdb8e7762ceb27bd196fcd9b..1b14dc8cbb4383bc1ca2f4395f2e75d6802fb949 100644
--- a/libhb/chroma_smooth.c
+++ b/libhb/chroma_smooth.c
@@ -8,7 +8,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #define CHROMA_SMOOTH_STRENGTH_DEFAULT 0.25
 #define CHROMA_SMOOTH_SIZE_DEFAULT 7
diff --git a/libhb/colormap.c b/libhb/colormap.c
index 071af4d88649a1a7ddf6ca70b2658ca3ebdafdf9..862036254458948050d42ec71b7fb3fe4f344f35 100644
--- a/libhb/colormap.c
+++ b/libhb/colormap.c
@@ -10,7 +10,7 @@
 #include <stdio.h>
 #include <stdint.h>
 #include <strings.h>
-#include "colormap.h"
+#include "handbrake/colormap.h"
 
 typedef struct
 {
diff --git a/libhb/colorspace.c b/libhb/colorspace.c
index 581886561a5369361dd90956a385d1f8cc4a3a3b..2e94fc0450b467dc736a7cb657b608d5c17475b0 100644
--- a/libhb/colorspace.c
+++ b/libhb/colorspace.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/avfilter_priv.h"
 
 static int colorspace_init(hb_filter_object_t * filter,
                            hb_filter_init_t * init);
diff --git a/libhb/comb_detect.c b/libhb/comb_detect.c
index 6bd9223982e2d928510b9619b1a9d3710a2f9429..b0917cb9c6059dfcde096bcafc38249d0cba8a54 100644
--- a/libhb/comb_detect.c
+++ b/libhb/comb_detect.c
@@ -28,8 +28,8 @@ Original "Faster" settings:
 #define FILTER_CLASSIC 1
 #define FILTER_ERODE_DILATE 2
 
-#include "handbrake.h"
-#include "taskset.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/taskset.h"
 
 typedef struct decomb_thread_arg_s {
     hb_filter_private_t *pv;
diff --git a/libhb/common.c b/libhb/common.c
index a3edef84a52ba67659178e64e10af077eca46a19..f7fb9dbb542e89f18f6d9c9709a75a4bbdd4efa5 100644
--- a/libhb/common.c
+++ b/libhb/common.c
@@ -12,15 +12,15 @@
 #include <ctype.h>
 #include <sys/time.h>
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "x264.h"
-#include "lang.h"
-#include "common.h"
-#include "h264_common.h"
-#include "h265_common.h"
-#include "encx264.h"
+#include "handbrake/lang.h"
+#include "handbrake/common.h"
+#include "handbrake/h264_common.h"
+#include "handbrake/h265_common.h"
+#include "handbrake/encx264.h"
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_common.h"
+#include "handbrake/qsv_common.h"
 #endif
 
 #if HB_PROJECT_FEATURE_X265
@@ -32,10 +32,10 @@
 #endif
 
 #if HB_PROJECT_FEATURE_NVENC
-#include "nvenc_common.h"
+#include "handbrake/nvenc_common.h"
 #endif
 #if HB_PROJECT_FEATURE_VCE
-#include "vce_common.h"
+#include "handbrake/vce_common.h"
 #endif
 
 #ifdef __APPLE__
diff --git a/libhb/compat.c b/libhb/compat.c
index 67a5dba7eb6adbdef76ce374d69846759156aa47..fd69dec8b9b314b3381589e72f7d28dd371cc25c 100644
--- a/libhb/compat.c
+++ b/libhb/compat.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "compat.h"
+#include "handbrake/compat.h"
 
 #ifdef HB_NEED_STRTOK_R
 #include <string.h>
diff --git a/libhb/cropscale.c b/libhb/cropscale.c
index 09af92e09e0d1c31e5336796b63f1e98d6ea8c39..10e7d01c96786de6a06183acfc9ff2e49faba5f5 100644
--- a/libhb/cropscale.c
+++ b/libhb/cropscale.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/avfilter_priv.h"
 
 static int crop_scale_init(hb_filter_object_t * filter,
                            hb_filter_init_t * init);
diff --git a/libhb/deblock.c b/libhb/deblock.c
index 31efc1d15954e870544566707ac4da4b2a225c9b..42627b48c899b892ed85bd92fd665e6f46b4a5d3 100644
--- a/libhb/deblock.c
+++ b/libhb/deblock.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/avfilter_priv.h"
 
 static int deblock_init(hb_filter_object_t * filter, hb_filter_init_t * init);
 
diff --git a/libhb/decavcodec.c b/libhb/decavcodec.c
index f705d7035b81dc11dd0119c7ea049736c70c9667..4c4fb34467307c420aca0c3a80359a6b9637027d 100644
--- a/libhb/decavcodec.c
+++ b/libhb/decavcodec.c
@@ -38,20 +38,20 @@
     VC1 decoder) can't easily be used by the HB mpeg stream reader.
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "hbavfilter.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/hbavfilter.h"
 #include "libavfilter/avfilter.h"
 #include "libavfilter/buffersrc.h"
 #include "libavfilter/buffersink.h"
 #include "libavutil/hwcontext.h"
-#include "lang.h"
-#include "audio_resample.h"
+#include "handbrake/lang.h"
+#include "handbrake/audio_resample.h"
 
 #if HB_PROJECT_FEATURE_QSV
 #include "libavutil/hwcontext_qsv.h"
-#include "qsv_common.h"
-#include "qsv_libav.h"
+#include "handbrake/qsv_common.h"
+#include "handbrake/qsv_libav.h"
 #endif
 
 static void compute_frame_duration( hb_work_private_t *pv );
diff --git a/libhb/deccc608sub.c b/libhb/deccc608sub.c
index e0defc7850e6e9e620d5fd6a498c334c6328551b..6a6a8494de8baf6202cebc0a7281783ad9359456 100644
--- a/libhb/deccc608sub.c
+++ b/libhb/deccc608sub.c
@@ -13,8 +13,8 @@
  *
  * Note that the SRT sub generation from CC could be useful for mkv subs.
  */
-#include "handbrake.h"
-#include "deccc608sub.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/deccc608sub.h"
 
 #define SSA_PREAMBLE_LEN 24
 /*
diff --git a/libhb/declpcm.c b/libhb/declpcm.c
index 1c322c568602274b443aef9b6e6408ce45070427..cf2f99de31f5f21ceb39ec0007d4488dfd4d6545 100644
--- a/libhb/declpcm.c
+++ b/libhb/declpcm.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "audio_resample.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/audio_resample.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/decomb.c b/libhb/decomb.c
index 13adfd01c6da00ddb6d3f535fdce79b21376e29b..3721f9a591b1cacda9da14e2b7c087fd985e33e9 100644
--- a/libhb/decomb.c
+++ b/libhb/decomb.c
@@ -54,11 +54,11 @@ which will feed EEDI2 interpolations to yadif.
 12-15: EEDI2 will override cubic interpolation
 *****/
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "eedi2.h"
-#include "taskset.h"
-#include "decomb.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/eedi2.h"
+#include "handbrake/taskset.h"
+#include "handbrake/decomb.h"
 
 #define PARITY_DEFAULT   -1
 
diff --git a/libhb/decpgssub.c b/libhb/decpgssub.c
index cafc4cf5c09128cb4df51257915e0a20ee597ed7..c7120484ac7559e808202094007bd5b5751a7319 100644
--- a/libhb/decpgssub.c
+++ b/libhb/decpgssub.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/decsrtsub.c b/libhb/decsrtsub.c
index 303387c329a00b850f946fe1efad991b6551667b..35aa9c1573ebbe1cb71fe990d4fb5f045fcbe5bb 100644
--- a/libhb/decsrtsub.c
+++ b/libhb/decsrtsub.c
@@ -12,9 +12,9 @@
 #include <string.h>
 #include <iconv.h>
 #include <errno.h>
-#include "handbrake.h"
-#include "colormap.h"
-#include "decsrtsub.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/colormap.h"
+#include "handbrake/decsrtsub.h"
 
 struct start_and_end {
     unsigned long start, end;
diff --git a/libhb/decssasub.c b/libhb/decssasub.c
index 9729e686325c7c47acf187ffcbf0ac096568ea16..b5d8339b97ede91a823012455ca528696ccc5769 100644
--- a/libhb/decssasub.c
+++ b/libhb/decssasub.c
@@ -26,11 +26,11 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #include <ass/ass.h>
-#include "decssasub.h"
-#include "colormap.h"
+#include "handbrake/decssasub.h"
+#include "handbrake/colormap.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/dectx3gsub.c b/libhb/dectx3gsub.c
index 49ea0e9b7cbbb7f8d4da93ff2e957fda4651e219..f2f5d4eea1dc54662ecaffd6795f67fca729921c 100644
--- a/libhb/dectx3gsub.c
+++ b/libhb/dectx3gsub.c
@@ -19,8 +19,8 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include "handbrake.h"
-#include "colormap.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/colormap.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/decutf8sub.c b/libhb/decutf8sub.c
index 45f522c22a43cb8398d6d61dce8d9a5ea4f22527..c3336d1aba13bd5dcec8e5c30ff1e69fef32965c 100644
--- a/libhb/decutf8sub.c
+++ b/libhb/decutf8sub.c
@@ -18,8 +18,8 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include "handbrake.h"
-#include "decsrtsub.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/decsrtsub.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/decvobsub.c b/libhb/decvobsub.c
index 4a6a6ec94394c3e36ce13d72175016df6d4fd219..be64a901380fcd47f6610632140c538983893a3d 100644
--- a/libhb/decvobsub.c
+++ b/libhb/decvobsub.c
@@ -27,7 +27,7 @@
  *   }
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/deinterlace.c b/libhb/deinterlace.c
index de91a82023fcec6c3f2bb250db12374b641ff97e..4ab1c1d16b74c7317e2da8800a7b2fe3496f51e1 100644
--- a/libhb/deinterlace.c
+++ b/libhb/deinterlace.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "decomb.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/decomb.h"
+#include "handbrake/avfilter_priv.h"
 
 static int deinterlace_init(hb_filter_object_t * filter,
                             hb_filter_init_t * init);
diff --git a/libhb/demuxmpeg.c b/libhb/demuxmpeg.c
index 09a9325b074eaaf0882e7964f9010485325b6f87..e765785861f449c6a1bfbf0a4426ada0a79e0254 100644
--- a/libhb/demuxmpeg.c
+++ b/libhb/demuxmpeg.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 static inline int check_mpeg_scr( hb_psdemux_t *state, int64_t scr, int tol )
 {
diff --git a/libhb/denoise.c b/libhb/denoise.c
index 0da2bf8af6cbc6a377fb2e864f8c533eafdd7d84..b43acaa84e759c6444ac70ce68daa04156995c8b 100644
--- a/libhb/denoise.c
+++ b/libhb/denoise.c
@@ -17,8 +17,8 @@
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 
 #define HQDN3D_SPATIAL_LUMA_DEFAULT    4.0f
 #define HQDN3D_SPATIAL_CHROMA_DEFAULT  3.0f
diff --git a/libhb/detelecine.c b/libhb/detelecine.c
index 2d2026124e05671478296c1cf55210c4be6cf95e..3d82838765423a5feb80f5e879ffb6d28e7ce830 100644
--- a/libhb/detelecine.c
+++ b/libhb/detelecine.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 
 /*
  *
diff --git a/libhb/dvd.c b/libhb/dvd.c
index 9b1c6fca0ec79cac14dce53fa697109d39b8ff5a..2ae506c7d1ac33bab0dc76a354a100c286a73473 100644
--- a/libhb/dvd.c
+++ b/libhb/dvd.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "lang.h"
-#include "dvd.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/lang.h"
+#include "handbrake/dvd.h"
 
 #include "dvdread/ifo_read.h"
 #include "dvdread/ifo_print.h"
diff --git a/libhb/dvdnav.c b/libhb/dvdnav.c
index 72833fcd4927959a6c0536f04ba94e4df65692df..52a48adfe79d87c1608478fe37479c764573bfa6 100644
--- a/libhb/dvdnav.c
+++ b/libhb/dvdnav.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "lang.h"
-#include "dvd.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/lang.h"
+#include "handbrake/dvd.h"
 
 #include "dvdnav/dvdnav.h"
 #include "dvdread/ifo_read.h"
diff --git a/libhb/eedi2.c b/libhb/eedi2.c
index e531ea96aab1d38b534edbb81884abd24f4e4751..92606183f2f49b5e97bb21bbb70bb350abedc835 100644
--- a/libhb/eedi2.c
+++ b/libhb/eedi2.c
@@ -10,8 +10,8 @@
    http://web.missouri.edu/~kes25c/
 */
 
-#include "handbrake.h"
-#include "eedi2.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/eedi2.h"
 
 /**
  * EEDI2 directional limit lookup table
diff --git a/libhb/enc_qsv.c b/libhb/enc_qsv.c
index 0b4f78a68163fe2b4e4fc80202be54122cbf890d..750bd35f23590be5cb29ddc742b71ac77f7a26f8 100644
--- a/libhb/enc_qsv.c
+++ b/libhb/enc_qsv.c
@@ -26,16 +26,16 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
-#include "handbrake.h"
-#include "nal_units.h"
-#include "qsv_common.h"
-#include "qsv_memory.h"
-#include "h264_common.h"
-#include "h265_common.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/nal_units.h"
+#include "handbrake/qsv_common.h"
+#include "handbrake/qsv_memory.h"
+#include "handbrake/h264_common.h"
+#include "handbrake/h265_common.h"
 
 #include "libavutil/hwcontext_qsv.h"
 #include "libavutil/hwcontext.h"
diff --git a/libhb/encavcodec.c b/libhb/encavcodec.c
index 99bf4010b493b15c8401ab53a828187371c9ae92..1c949249c4e93ceef0ca3601c0bce9f3051c24a9 100644
--- a/libhb/encavcodec.c
+++ b/libhb/encavcodec.c
@@ -7,12 +7,12 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hb_dict.h"
-#include "hbffmpeg.h"
-#include "h264_common.h"
-#include "h265_common.h"
-#include "nal_units.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/h264_common.h"
+#include "handbrake/h265_common.h"
+#include "handbrake/nal_units.h"
 
 /*
  * The frame info struct remembers information about each frame across calls
diff --git a/libhb/encavcodecaudio.c b/libhb/encavcodecaudio.c
index 29117ded61b0bccd048050c6a502a13f53ec8201..7c81d1736b01636bf75431b6a53cf75984408d77 100644
--- a/libhb/encavcodecaudio.c
+++ b/libhb/encavcodecaudio.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/enctheora.c b/libhb/enctheora.c
index 429f8e2fa8b9e3a492ba4ae56d90b8b99943ee0b..7a3f969c8252d16e5384899600d4f787af3f6969 100644
--- a/libhb/enctheora.c
+++ b/libhb/enctheora.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "theora/codec.h"
 #include "theora/theoraenc.h"
 
diff --git a/libhb/encvobsub.c b/libhb/encvobsub.c
index 0650f3501e684550693a9111aee7f81645b0f4ab..33b611ed0d568557faa3e0f454d778bb82829aa0 100644
--- a/libhb/encvobsub.c
+++ b/libhb/encvobsub.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 struct hb_work_private_s
 {
diff --git a/libhb/encvorbis.c b/libhb/encvorbis.c
index 3d521e3eab2251d934efca9f815e26a953906962..92f24a13f3eca67c55043f499e85b977b3cba35e 100644
--- a/libhb/encvorbis.c
+++ b/libhb/encvorbis.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "audio_remap.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/audio_remap.h"
 
 #include "vorbis/vorbisenc.h"
 
diff --git a/libhb/encx264.c b/libhb/encx264.c
index b0f6781bc45456fd07d132b2572d17c12e0de495..a94eb066e3e594385ae67b73ee48b5cff6333452 100644
--- a/libhb/encx264.c
+++ b/libhb/encx264.c
@@ -9,9 +9,9 @@
 
 #include <stdarg.h>
 
-#include "handbrake.h"
-#include "hb_dict.h"
-#include "encx264.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/encx264.h"
 
 int  encx264Init( hb_work_object_t *, hb_job_t * );
 int  encx264Work( hb_work_object_t *, hb_buffer_t **, hb_buffer_t ** );
diff --git a/libhb/encx265.c b/libhb/encx265.c
index 39e912737e5f7911131f349b0fdeea544a531792..025f80120742feef0614f4b693a04de4f0a17c1f 100644
--- a/libhb/encx265.c
+++ b/libhb/encx265.c
@@ -7,13 +7,13 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_X265
 
-#include "handbrake.h"
-#include "hb_dict.h"
-#include "h265_common.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/h265_common.h"
 #include "x265.h"
 
 int  encx265Init (hb_work_object_t*, hb_job_t*);
diff --git a/libhb/fifo.c b/libhb/fifo.c
index 1bc89f82359bc7256e3ede2fb0310dbab1b2f5bd..1ded249fc782402b36a36b33e6d953e6ac8f94f5 100644
--- a/libhb/fifo.c
+++ b/libhb/fifo.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_libav.h"
+#include "handbrake/qsv_libav.h"
 #endif
 
 #ifndef SYS_DARWIN
diff --git a/libhb/grayscale.c b/libhb/grayscale.c
index 1608f8f453202fed51c8feb4ff3300576da3b806..d8de86224d87b9d89adb93148e82aecb850143cb 100644
--- a/libhb/grayscale.c
+++ b/libhb/grayscale.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "taskset.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/taskset.h"
 
 // Settings:
 //  This filter has no settings.
diff --git a/libhb/audio_remap.h b/libhb/handbrake/audio_remap.h
similarity index 100%
rename from libhb/audio_remap.h
rename to libhb/handbrake/audio_remap.h
diff --git a/libhb/audio_resample.h b/libhb/handbrake/audio_resample.h
similarity index 100%
rename from libhb/audio_resample.h
rename to libhb/handbrake/audio_resample.h
diff --git a/libhb/avfilter_priv.h b/libhb/handbrake/avfilter_priv.h
similarity index 96%
rename from libhb/avfilter_priv.h
rename to libhb/handbrake/avfilter_priv.h
index 8c6b0a62734b8d6f43b96ac431357e52a53a795d..20e616cc236b9b837c6da2e1a290f54900a25067 100644
--- a/libhb/avfilter_priv.h
+++ b/libhb/handbrake/avfilter_priv.h
@@ -11,7 +11,7 @@
 #define HB_AVFILTER_PRIV_H
 
 #include "libavfilter/avfilter.h"
-#include "hbavfilter.h"
+#include "handbrake/hbavfilter.h"
 
 struct hb_filter_private_s
 {
diff --git a/libhb/bits.h b/libhb/handbrake/bits.h
similarity index 100%
rename from libhb/bits.h
rename to libhb/handbrake/bits.h
diff --git a/libhb/colormap.h b/libhb/handbrake/colormap.h
similarity index 100%
rename from libhb/colormap.h
rename to libhb/handbrake/colormap.h
diff --git a/libhb/common.h b/libhb/handbrake/common.h
similarity index 99%
rename from libhb/common.h
rename to libhb/handbrake/common.h
index cd66d6f1604c55e4959b9422a73b305aa24c0159..8ab21fa8abc6933f58aa221d0fcbbd14b53f0599 100644
--- a/libhb/common.h
+++ b/libhb/handbrake/common.h
@@ -10,9 +10,9 @@
 #ifndef HB_COMMON_H
 #define HB_COMMON_H
 
-#include "project.h"
-#include "hbtypes.h"
-#include "hb_dict.h"
+#include "handbrake/project.h"
+#include "handbrake/hbtypes.h"
+#include "handbrake/hb_dict.h"
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
diff --git a/libhb/compat.h b/libhb/handbrake/compat.h
similarity index 100%
rename from libhb/compat.h
rename to libhb/handbrake/compat.h
diff --git a/libhb/deccc608sub.h b/libhb/handbrake/deccc608sub.h
similarity index 99%
rename from libhb/deccc608sub.h
rename to libhb/handbrake/deccc608sub.h
index 14f98c25d9403ea21b601fefdb39bf8613d32f0f..044fc78f763aa9f95c65a43972a26b8303619807 100644
--- a/libhb/deccc608sub.h
+++ b/libhb/handbrake/deccc608sub.h
@@ -13,7 +13,7 @@
 #ifndef __DECCC608SUB_H__
 #define __DECCC608SUB_H__
 
-#include "common.h"
+#include "handbrake/common.h"
 
 struct s_write;
 
diff --git a/libhb/decomb.h b/libhb/handbrake/decomb.h
similarity index 100%
rename from libhb/decomb.h
rename to libhb/handbrake/decomb.h
diff --git a/libhb/decsrtsub.h b/libhb/handbrake/decsrtsub.h
similarity index 100%
rename from libhb/decsrtsub.h
rename to libhb/handbrake/decsrtsub.h
diff --git a/libhb/decssasub.h b/libhb/handbrake/decssasub.h
similarity index 100%
rename from libhb/decssasub.h
rename to libhb/handbrake/decssasub.h
diff --git a/libhb/dvd.h b/libhb/handbrake/dvd.h
similarity index 100%
rename from libhb/dvd.h
rename to libhb/handbrake/dvd.h
diff --git a/libhb/eedi2.h b/libhb/handbrake/eedi2.h
similarity index 100%
rename from libhb/eedi2.h
rename to libhb/handbrake/eedi2.h
diff --git a/libhb/encx264.h b/libhb/handbrake/encx264.h
similarity index 98%
rename from libhb/encx264.h
rename to libhb/handbrake/encx264.h
index 75b98675bc2ba818efc159b6d48327d800f3b42c..29085e50cf7fb5c5e8f12d02e56105466973f852 100644
--- a/libhb/encx264.h
+++ b/libhb/handbrake/encx264.h
@@ -11,7 +11,7 @@
 #define HB_ENCX264_H
 
 #include "x264.h"
-#include "h264_common.h"
+#include "handbrake/h264_common.h"
 
 /* x264 preferred option names (left) and synonyms (right).
  * The "preferred" names match names used in x264's param2string function more
diff --git a/libhb/h264_common.h b/libhb/handbrake/h264_common.h
similarity index 100%
rename from libhb/h264_common.h
rename to libhb/handbrake/h264_common.h
diff --git a/libhb/h265_common.h b/libhb/handbrake/h265_common.h
similarity index 98%
rename from libhb/h265_common.h
rename to libhb/handbrake/h265_common.h
index 067b652e5c442a60221d9679b8db16b7996892fd..8b5c1072dece6415a8f39b1fe16c31ec18b64528 100644
--- a/libhb/h265_common.h
+++ b/libhb/handbrake/h265_common.h
@@ -10,7 +10,7 @@
 #ifndef HB_H265_COMMON_H
 #define HB_H265_COMMON_H
 
-#include "project.h"
+#include "handbrake/project.h"
 
 // inspired by libavcodec/hevc.h
 // in HEVC, all "random access point" NAL units are keyframes
diff --git a/libhb/handbrake.h b/libhb/handbrake/handbrake.h
similarity index 95%
rename from libhb/handbrake.h
rename to libhb/handbrake/handbrake.h
index 9cf7444762d0daf3b45626e0f55276cbd751cd0e..6143316d56cafe2e1c9960c04cca21f94d50dc82 100644
--- a/libhb/handbrake.h
+++ b/libhb/handbrake/handbrake.h
@@ -14,14 +14,14 @@
 extern "C" {
 #endif
 
-#include "common.h"
-#include "project.h"
-#include "compat.h"
-#include "hb_json.h"
-#include "preset.h"
-#include "plist.h"
-#include "param.h"
-#include "colormap.h"
+#include "handbrake/common.h"
+#include "handbrake/project.h"
+#include "handbrake/compat.h"
+#include "handbrake/hb_json.h"
+#include "handbrake/preset.h"
+#include "handbrake/plist.h"
+#include "handbrake/param.h"
+#include "handbrake/colormap.h"
 
 /* hb_init()
    Initializes a libhb session (launches his own thread, detects CPUs,
diff --git a/libhb/hb_dict.h b/libhb/handbrake/hb_dict.h
similarity index 99%
rename from libhb/hb_dict.h
rename to libhb/handbrake/hb_dict.h
index 3984d1a0d8e6537b739f29fef90b3f240719ecb6..5d8a188e02ee2039eb720fb8c8aa27af0ec4c19e 100644
--- a/libhb/hb_dict.h
+++ b/libhb/handbrake/hb_dict.h
@@ -9,7 +9,7 @@
 #if !defined(HB_DICT_H)
 #define HB_DICT_H
 
-#include "hbtypes.h"
+#include "handbrake/hbtypes.h"
 #include <jansson.h>
 
 #define HB_VALUE_TYPE_DICT      JSON_OBJECT
diff --git a/libhb/hb_json.h b/libhb/handbrake/hb_json.h
similarity index 98%
rename from libhb/hb_json.h
rename to libhb/handbrake/hb_json.h
index 8f8a08cbcf3cdcba31ad784a1e5f2e6b42758a41..aac3223f75e5a7b31c6b63f4d594dc167545b842 100644
--- a/libhb/hb_json.h
+++ b/libhb/handbrake/hb_json.h
@@ -14,7 +14,7 @@
 extern "C" {
 #endif
 
-#include "common.h"
+#include "handbrake/common.h"
 
 hb_dict_t  * hb_state_to_dict( hb_state_t * state);
 hb_dict_t  * hb_job_to_dict( const hb_job_t * job );
diff --git a/libhb/hbavfilter.h b/libhb/handbrake/hbavfilter.h
similarity index 97%
rename from libhb/hbavfilter.h
rename to libhb/handbrake/hbavfilter.h
index 1495d3b41b300e4288637aacb2b64e8bad251dac..319b4887142a6e9abed728f9ec803bc5d50c6066 100644
--- a/libhb/hbavfilter.h
+++ b/libhb/handbrake/hbavfilter.h
@@ -10,7 +10,7 @@
 #ifndef HB_AVFILTER_H
 #define HB_AVFILTER_H
 
-#include "common.h"
+#include "handbrake/common.h"
 
 typedef struct hb_avfilter_graph_s hb_avfilter_graph_t;
 
diff --git a/libhb/hbffmpeg.h b/libhb/handbrake/hbffmpeg.h
similarity index 98%
rename from libhb/hbffmpeg.h
rename to libhb/handbrake/hbffmpeg.h
index 119347153a30caa1ae15fdd48a4fd04a6d815935..6357dcd9c2c9796e393feeb4b032238741a2ef37 100644
--- a/libhb/hbffmpeg.h
+++ b/libhb/handbrake/hbffmpeg.h
@@ -21,7 +21,7 @@
 #include "libavutil/display.h"
 #include "libswscale/swscale.h"
 #include "libswresample/swresample.h"
-#include "common.h"
+#include "handbrake/common.h"
 
 #define HB_FFMPEG_THREADS_AUTO (-1) // let hb_avcodec_open() decide thread_count
 
diff --git a/libhb/hbtypes.h b/libhb/handbrake/hbtypes.h
similarity index 100%
rename from libhb/hbtypes.h
rename to libhb/handbrake/hbtypes.h
diff --git a/libhb/internal.h b/libhb/handbrake/internal.h
similarity index 99%
rename from libhb/internal.h
rename to libhb/handbrake/internal.h
index e642c2d2c255c6bae490bd02b39fa4e96a62c358..93ae27dca74e72834fc45dadee21011ce73ca69a 100644
--- a/libhb/internal.h
+++ b/libhb/handbrake/internal.h
@@ -10,10 +10,10 @@
 #ifndef HB_INTERNAL_H
 #define HB_INTERNAL_H
 
-#include "project.h"
-#include "hbffmpeg.h"
+#include "handbrake/project.h"
+#include "handbrake/hbffmpeg.h"
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_libav.h"
+#include "handbrake/qsv_libav.h"
 #endif
 
 /***********************************************************************
diff --git a/libhb/lang.h b/libhb/handbrake/lang.h
similarity index 100%
rename from libhb/lang.h
rename to libhb/handbrake/lang.h
diff --git a/libhb/nal_units.h b/libhb/handbrake/nal_units.h
similarity index 98%
rename from libhb/nal_units.h
rename to libhb/handbrake/nal_units.h
index 21041904152f36cb37e92a97d764ad2dec84a7e5..0efccc4763cf382a508a838c317d42005fbce600 100644
--- a/libhb/nal_units.h
+++ b/libhb/handbrake/nal_units.h
@@ -12,7 +12,7 @@
 
 #include <stdint.h>
 
-#include "common.h"
+#include "handbrake/common.h"
 
 /*
  * Write a NAL unit of the specified size to the provided
diff --git a/libhb/nlmeans.h b/libhb/handbrake/nlmeans.h
similarity index 100%
rename from libhb/nlmeans.h
rename to libhb/handbrake/nlmeans.h
diff --git a/libhb/nvenc_common.h b/libhb/handbrake/nvenc_common.h
similarity index 100%
rename from libhb/nvenc_common.h
rename to libhb/handbrake/nvenc_common.h
diff --git a/libhb/param.h b/libhb/handbrake/param.h
similarity index 100%
rename from libhb/param.h
rename to libhb/handbrake/param.h
diff --git a/libhb/plist.h b/libhb/handbrake/plist.h
similarity index 95%
rename from libhb/plist.h
rename to libhb/handbrake/plist.h
index e33d7fc3300c8db4a8b8fd42bf9cd1cdd62580e2..12b51dd2e3bddcc9df45a7800e6a6ec11af3ee02 100644
--- a/libhb/plist.h
+++ b/libhb/handbrake/plist.h
@@ -11,7 +11,7 @@
 #define _HB_PLIST_H_
 
 #include <stdio.h>
-#include "hb_dict.h"
+#include "handbrake/hb_dict.h"
 
 hb_value_t * hb_plist_parse(const char *buf, size_t len);
 hb_value_t * hb_plist_parse_file(const char *filename);
diff --git a/libhb/ports.h b/libhb/handbrake/ports.h
similarity index 99%
rename from libhb/ports.h
rename to libhb/handbrake/ports.h
index e1eef623d3a5ce7975dbbc87ce33b07806dfe43f..6b793da28210c69a9ff30f4eebebbc4821501fc4 100644
--- a/libhb/ports.h
+++ b/libhb/handbrake/ports.h
@@ -24,7 +24,7 @@
 #define IS_DIR_SEP(c) (c == '/')
 #endif
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 #include "mfx/mfxstructures.h"
diff --git a/libhb/preset.h b/libhb/handbrake/preset.h
similarity index 99%
rename from libhb/preset.h
rename to libhb/handbrake/preset.h
index fda7ebefdcee730ecbd6853e7c4cf21146a8ef19..abb188487b8f3574b920a50b295b1d579051a965 100644
--- a/libhb/preset.h
+++ b/libhb/handbrake/preset.h
@@ -9,8 +9,8 @@
 #if !defined(HB_PRESET_H)
 #define HB_PRESET_H
 
-#include "common.h"
-#include "hb_dict.h"
+#include "handbrake/common.h"
+#include "handbrake/hb_dict.h"
 
 #define HB_MAX_PRESET_FOLDER_DEPTH  8
 
diff --git a/libhb/preset_builtin.h b/libhb/handbrake/preset_builtin.h
similarity index 100%
rename from libhb/preset_builtin.h
rename to libhb/handbrake/preset_builtin.h
diff --git a/libhb/project.h.m4 b/libhb/handbrake/project.h.m4
similarity index 100%
rename from libhb/project.h.m4
rename to libhb/handbrake/project.h.m4
diff --git a/libhb/qsv_common.h b/libhb/handbrake/qsv_common.h
similarity index 99%
rename from libhb/qsv_common.h
rename to libhb/handbrake/qsv_common.h
index 9522f74881079db91bdcbb16bbb51334afb86144..ca844c5e9e983fc331a4ba469aced6adccaef9d4 100644
--- a/libhb/qsv_common.h
+++ b/libhb/handbrake/qsv_common.h
@@ -12,14 +12,14 @@
 
 int            hb_qsv_available();
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
 #include "mfx/mfxvideo.h"
 #include "mfx/mfxplugin.h"
 #include "libavcodec/avcodec.h"
-#include "hb_dict.h"
+#include "handbrake/hb_dict.h"
 
 /* Minimum Intel Media SDK version (currently 1.3, for Sandy Bridge support) */
 #define HB_QSV_MINVERSION_MAJOR HB_QSV_MSDK_VERSION_MAJOR
diff --git a/libhb/qsv_filter.h b/libhb/handbrake/qsv_filter.h
similarity index 98%
rename from libhb/qsv_filter.h
rename to libhb/handbrake/qsv_filter.h
index fc3ba95f88cdc7a0caf9202c7b36e4eb281c9376..6d2650059b2ef16cee971ad4e0366377f1faa86f 100644
--- a/libhb/qsv_filter.h
+++ b/libhb/handbrake/qsv_filter.h
@@ -29,7 +29,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef QSV_FILTER_H
 #define QSV_FILTER_H
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 void qsv_filter_close( hb_qsv_context* qsv, HB_QSV_STAGE_TYPE vpp_type );
diff --git a/libhb/qsv_filter_pp.h b/libhb/handbrake/qsv_filter_pp.h
similarity index 99%
rename from libhb/qsv_filter_pp.h
rename to libhb/handbrake/qsv_filter_pp.h
index bb2c3cc68afb8e1bcea8f9fc5049bc8f9337528b..c0e7f605e5623cd467a0d9c342e17f9baf8d328a 100644
--- a/libhb/qsv_filter_pp.h
+++ b/libhb/handbrake/qsv_filter_pp.h
@@ -29,7 +29,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef QSV_FILTER_PP_H
 #define QSV_FILTER_PP_H
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
diff --git a/libhb/qsv_libav.h b/libhb/handbrake/qsv_libav.h
similarity index 100%
rename from libhb/qsv_libav.h
rename to libhb/handbrake/qsv_libav.h
diff --git a/libhb/qsv_memory.h b/libhb/handbrake/qsv_memory.h
similarity index 97%
rename from libhb/qsv_memory.h
rename to libhb/handbrake/qsv_memory.h
index 9397306f069dc43afec60ca6ea658f43e625e848..3434114e17e80658016799ab6a37a15015c4147f 100644
--- a/libhb/qsv_memory.h
+++ b/libhb/handbrake/qsv_memory.h
@@ -29,12 +29,12 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef QSV_MEMORY_H
 #define QSV_MEMORY_H
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
 #include "mfx/mfxplugin.h"
-#include "qsv_libav.h"
+#include "handbrake/qsv_libav.h"
 
 typedef struct{
 
diff --git a/libhb/ssautil.h b/libhb/handbrake/ssautil.h
similarity index 100%
rename from libhb/ssautil.h
rename to libhb/handbrake/ssautil.h
diff --git a/libhb/taskset.h b/libhb/handbrake/taskset.h
similarity index 98%
rename from libhb/taskset.h
rename to libhb/handbrake/taskset.h
index 07a7e32bc2b7da6eb9439fe55a64c97c634d61e5..35c21e1c000be273142d8dbcdd60a26f8142218e 100644
--- a/libhb/taskset.h
+++ b/libhb/handbrake/taskset.h
@@ -12,7 +12,7 @@
 
 #define TASKSET_POSIX_COMPLIANT 1
 
-#include "bits.h"
+#include "handbrake/bits.h"
 
 typedef struct hb_taskset_s {
     int                thread_count;
diff --git a/libhb/vce_common.h b/libhb/handbrake/vce_common.h
similarity index 100%
rename from libhb/vce_common.h
rename to libhb/handbrake/vce_common.h
diff --git a/libhb/hb.c b/libhb/hb.c
index b44c2679efa4cbc29ea033657aaf9e4254b99a31..da65650a13cadc506ce6706b6ee7fb1b42b1f8cb 100644
--- a/libhb/hb.c
+++ b/libhb/hb.c
@@ -7,16 +7,16 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "encx264.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/encx264.h"
 #include "libavfilter/avfilter.h"
 #include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
 
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_common.h"
+#include "handbrake/qsv_common.h"
 #endif
 
 #if defined( SYS_MINGW )
diff --git a/libhb/hb_dict.c b/libhb/hb_dict.c
index b9bd5a8d3356c1c086cb6dbbeee9aee952d2a034..52f585a94da6c74b4461cff9508866bc110ca927 100644
--- a/libhb/hb_dict.c
+++ b/libhb/hb_dict.c
@@ -9,8 +9,8 @@
 
 #include <ctype.h>
 #include <stdio.h>
-#include "handbrake.h"
-#include "hb_dict.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hb_dict.h"
 
 hb_value_type_t hb_value_type(const hb_value_t *value)
 {
diff --git a/libhb/hb_json.c b/libhb/hb_json.c
index 5fb92b8b8056d9e5514596f49137448bd8f30f17..8ad0a720dd16d6950711c393b6542b785010fa5b 100644
--- a/libhb/hb_json.c
+++ b/libhb/hb_json.c
@@ -8,10 +8,10 @@
  */
 
 #include <jansson.h>
-#include "handbrake.h"
-#include "hb_json.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hb_json.h"
 #include "libavutil/base64.h"
-#include "qsv_common.h"
+#include "handbrake/qsv_common.h"
 
 /**
  * Convert an hb_state_t to a jansson dict
diff --git a/libhb/hbavfilter.c b/libhb/hbavfilter.c
index 5e050e28f070b69a73d2e91b4a94c9bb8d022e7f..fa86ac5f8cec5a26745cc95b58ad12dbc1285059 100644
--- a/libhb/hbavfilter.c
+++ b/libhb/hbavfilter.c
@@ -7,13 +7,13 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 #include "libavfilter/avfilter.h"
 #include "libavfilter/buffersrc.h"
 #include "libavfilter/buffersink.h"
-#include "hbavfilter.h"
-#include "avfilter_priv.h"
+#include "handbrake/hbavfilter.h"
+#include "handbrake/avfilter_priv.h"
 
 struct hb_avfilter_graph_s
 {
diff --git a/libhb/hbffmpeg.c b/libhb/hbffmpeg.c
index af0707874becc0c720d988aae1ab78782cc504cb..f5a140c754c06b7a5510ee2662defa4cca7f7157 100644
--- a/libhb/hbffmpeg.c
+++ b/libhb/hbffmpeg.c
@@ -1,5 +1,5 @@
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 
 static int get_frame_type(int type)
 {
diff --git a/libhb/lang.c b/libhb/lang.c
index cd3c8887fbf2bc424dbdadfe446dc6e6eb7a880f..0cf828a61a6aeb4bbe3935fb499d7975d5d811fb 100644
--- a/libhb/lang.c
+++ b/libhb/lang.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "lang.h"
+#include "handbrake/lang.h"
 #include <string.h>
 #include <ctype.h>
 
diff --git a/libhb/lapsharp.c b/libhb/lapsharp.c
index 192b26721f4ee49adc5b374eb42644fdc37f6831..9174ac39492d59fd5dbc357119acc2210d68343d 100644
--- a/libhb/lapsharp.c
+++ b/libhb/lapsharp.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #define LAPSHARP_STRENGTH_LUMA_DEFAULT   0.2
 #define LAPSHARP_STRENGTH_CHROMA_DEFAULT 0.2
diff --git a/libhb/module.defs b/libhb/module.defs
index e76358f396861859f189ab489fa18038fecfaa36..a2d444447623996b6a1352359cb5175900ae15fd 100644
--- a/libhb/module.defs
+++ b/libhb/module.defs
@@ -15,7 +15,7 @@ $(eval $(call import.GCC,LIBHB))
 LIBHB.src/   = $(SRC/)libhb/
 LIBHB.build/ = $(BUILD/)libhb/
 
-LIBHB.m4.in  = $(wildcard $(LIBHB.src/)*.m4)
+LIBHB.m4.in  = $(wildcard $(LIBHB.src/)handbrake/*.m4)
 LIBHB.m4.out = $(patsubst $(LIBHB.src/)%.m4,$(LIBHB.build/)%,$(LIBHB.m4.in))
 
 LIBHB.c   = $(wildcard $(LIBHB.src/)*.c)
@@ -24,7 +24,7 @@ LIBHB.m.o = $(patsubst $(SRC/)%.m,$(BUILD/)%.o,$(LIBHB.m))
 LIBHB.d   = $(LIBHB.m4.out) $(LIBHB.h.out) \
     $(foreach n,$(LIBHB.prerequisites),$($n.INSTALL.target) )
 
-LIBHB.h.in  = $(wildcard $(LIBHB.src/)*.h)
+LIBHB.h.in  = $(wildcard $(LIBHB.src/)handbrake/*.h)
 LIBHB.h.in += $(wildcard $(LIBHB.src/)extras/*.h)
 LIBHB.h.out = $(patsubst $(SRC/)%,$(BUILD/)%,$(LIBHB.h.in))
 
diff --git a/libhb/mt_frame_filter.c b/libhb/mt_frame_filter.c
index fd6bf780e9ad0b463e70a2170cd793bbbc2eae04..11f5ffc72946b9d829c4b1b41ee6c53b95fe3829 100644
--- a/libhb/mt_frame_filter.c
+++ b/libhb/mt_frame_filter.c
@@ -12,8 +12,8 @@
  * operate on each frame independently with no context carried over
  * from one frame to the next. */
 
-#include "handbrake.h"
-#include "taskset.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/taskset.h"
 
 typedef struct
 {
diff --git a/libhb/muxavformat.c b/libhb/muxavformat.c
index 1d630ebd25c4d8dc8bb441efc87d48683b4424e6..983c5d3489b68762a9e5ffdc9df08e71cff1f9a6 100644
--- a/libhb/muxavformat.c
+++ b/libhb/muxavformat.c
@@ -12,9 +12,9 @@
 #include "libavutil/avstring.h"
 #include "libavutil/intreadwrite.h"
 
-#include "handbrake.h"
-#include "ssautil.h"
-#include "lang.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/ssautil.h"
+#include "handbrake/lang.h"
 
 struct hb_mux_data_s
 {
diff --git a/libhb/muxcommon.c b/libhb/muxcommon.c
index a16a27a5216aae81e5a0e2d755141f0c142b9ec1..842ed86e8dc045a10ec1471c03a7b89be7cd2000 100644
--- a/libhb/muxcommon.c
+++ b/libhb/muxcommon.c
@@ -6,8 +6,8 @@
    It may be used under the terms of the GNU General Public License v2.
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
-#include "handbrake.h"
-#include "decssasub.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/decssasub.h"
 
 #define MIN_BUFFERING (1024*1024*10)
 #define MAX_BUFFERING (1024*1024*50)
diff --git a/libhb/nal_units.c b/libhb/nal_units.c
index 5fd332a37779dc39218cbb57208a2ba4e9a59f79..9c326b81a7a0cc8266120e4b3e33ceaddf0816ac 100644
--- a/libhb/nal_units.c
+++ b/libhb/nal_units.c
@@ -10,8 +10,8 @@
 #include <stdint.h>
 #include <string.h>
 
-#include "common.h"
-#include "nal_units.h"
+#include "handbrake/common.h"
+#include "handbrake/nal_units.h"
 
 static const uint8_t hb_annexb_startcode[] = { 0x00, 0x00, 0x00, 0x01, };
 
diff --git a/libhb/nlmeans.c b/libhb/nlmeans.c
index d838e057563cec81c1d79596efc5bcfafbde8b56..2297d4f7be61374b4c32878315e655b2246b719e 100644
--- a/libhb/nlmeans.c
+++ b/libhb/nlmeans.c
@@ -50,10 +50,10 @@
  *        etc...
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "taskset.h"
-#include "nlmeans.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/taskset.h"
+#include "handbrake/nlmeans.h"
 
 #define NLMEANS_STRENGTH_LUMA_DEFAULT      6
 #define NLMEANS_STRENGTH_CHROMA_DEFAULT    6
diff --git a/libhb/nlmeans_x86.c b/libhb/nlmeans_x86.c
index 131b8100793f6a50a583f60fd6832dfaee5387ac..423fa50e43f5d42e01f9c0769656c916c862f1bf 100644
--- a/libhb/nlmeans_x86.c
+++ b/libhb/nlmeans_x86.c
@@ -8,14 +8,14 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"     // needed for ARCH_X86
+#include "handbrake/handbrake.h"     // needed for ARCH_X86
 
 #if defined(ARCH_X86)
 
 #include <emmintrin.h>
 
 #include "libavutil/cpu.h"
-#include "nlmeans.h"
+#include "handbrake/nlmeans.h"
 
 static void build_integral_sse2(uint32_t *integral,
                                 int       integral_stride,
diff --git a/libhb/nvenc_common.c b/libhb/nvenc_common.c
index fb711b0ed540b757f98d8ddff0a510e8a6f1a6b9..c6897a84c8869a7f287abcf4f0647573d36a6a4c 100644
--- a/libhb/nvenc_common.c
+++ b/libhb/nvenc_common.c
@@ -7,8 +7,8 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "hbffmpeg.h"
-#include "handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/handbrake.h"
 
 #if HB_PROJECT_FEATURE_NVENC
 #include <ffnvcodec/nvEncodeAPI.h>
diff --git a/libhb/pad.c b/libhb/pad.c
index 10631bffe7771008844dc8a86623ec39366d361b..eee3e1c79ff70a6ee6f682e2f80446b5d9ca7eaf 100644
--- a/libhb/pad.c
+++ b/libhb/pad.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "colormap.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/colormap.h"
+#include "handbrake/avfilter_priv.h"
 
 static int pad_init(hb_filter_object_t * filter, hb_filter_init_t * init);
 
diff --git a/libhb/param.c b/libhb/param.c
index 1f73de62243e4a4804fc27a29512dd015485cc8a..e5baa9d26a385896cb82386abda1e1311fbe2767 100644
--- a/libhb/param.c
+++ b/libhb/param.c
@@ -8,13 +8,13 @@
  * http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "project.h"
-#include "hb_dict.h"
-#include "param.h"
-#include "common.h"
-#include "colormap.h"
+#include "handbrake/project.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/param.h"
+#include "handbrake/common.h"
+#include "handbrake/colormap.h"
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_common.h"
+#include "handbrake/qsv_common.h"
 #endif
 #include <regex.h>
 
diff --git a/libhb/platform/macosx/encca_aac.c b/libhb/platform/macosx/encca_aac.c
index 8d770c2e9e4c1cfcaef4f55932f221f0daea7a13..5c108cba436c997a642efa78a8d3be0b7a085d0f 100644
--- a/libhb/platform/macosx/encca_aac.c
+++ b/libhb/platform/macosx/encca_aac.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "audio_remap.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/audio_remap.h"
 #include <AudioToolbox/AudioToolbox.h>
 #include <CoreAudio/CoreAudio.h>
 
diff --git a/libhb/platform/macosx/vt_common.c b/libhb/platform/macosx/vt_common.c
index 51c0ff97c6e57b6128e81e7230ea592f21a4082d..3ea9ce15e854faf22530cead1ceef036b9c5c0bd 100644
--- a/libhb/platform/macosx/vt_common.c
+++ b/libhb/platform/macosx/vt_common.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "vt_common.h"
 
 #include <VideoToolbox/VideoToolbox.h>
diff --git a/libhb/plist.c b/libhb/plist.c
index 6db6dd389de4b7ecfbdf63d862cf97bbef4860f3..2bd5a02ec05412f73561a43dcf5e0bf654f1f168 100644
--- a/libhb/plist.c
+++ b/libhb/plist.c
@@ -14,9 +14,9 @@
 #include <inttypes.h>
 #include "libxml/parser.h"
 
-#include "common.h"
-#include "hb_dict.h"
-#include "plist.h"
+#include "handbrake/common.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/plist.h"
 
 #define BUF_SZ  (128*1024)
 
diff --git a/libhb/ports.c b/libhb/ports.c
index 6086aa628d35c18d1bc2571c6279ffac809236eb..917da62b9807b65cfc7d5eb8fa3015760f28619f 100644
--- a/libhb/ports.c
+++ b/libhb/ports.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #ifdef SYS_MINGW
 #define _WIN32_WINNT 0x600
@@ -88,7 +88,7 @@
 #include <stddef.h>
 #include <unistd.h>
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "libavutil/cpu.h"
 
 /************************************************************************
diff --git a/libhb/preset.c b/libhb/preset.c
index b592a2bbaa4ba99453ad0e9907bb1848350524b0..b7a30013a1242f4bddc92df7e1fbc7502f002837 100644
--- a/libhb/preset.c
+++ b/libhb/preset.c
@@ -7,14 +7,14 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "preset_builtin.h"
-#include "handbrake.h"
-#include "hb_dict.h"
-#include "plist.h"
-#include "lang.h"
+#include "handbrake/preset_builtin.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/plist.h"
+#include "handbrake/lang.h"
 
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_common.h"
+#include "handbrake/qsv_common.h"
 #endif
 
 #if defined(SYS_LINUX)
diff --git a/libhb/qsv_common.c b/libhb/qsv_common.c
index 9f7762159520966bc0131dfd6b2413da787b39c7..08d4f030b6d07033053bde51bbdad33afb34f279 100644
--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -7,20 +7,26 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
 #include <stdio.h>
 #include <string.h>
 
-#include "handbrake.h"
-#include "ports.h"
-#include "common.h"
-#include "hb_dict.h"
-#include "qsv_common.h"
-#include "h264_common.h"
-#include "h265_common.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/ports.h"
+#include "handbrake/common.h"
+#include "handbrake/hb_dict.h"
+#include "handbrake/qsv_common.h"
+#include "handbrake/h264_common.h"
+#include "handbrake/h265_common.h"
+#include "handbrake/hbffmpeg.h"
+#include "libavfilter/avfilter.h"
+#include "libavfilter/buffersrc.h"
+#include "libavfilter/buffersink.h"
+#include "libavutil/hwcontext_qsv.h"
+#include "libavutil/hwcontext.h"
 
 // QSV info for each codec
 static hb_qsv_info_t *hb_qsv_info_avc       = NULL;
@@ -2279,16 +2285,6 @@ void hb_qsv_force_workarounds()
 #undef FORCE_WORKAROUNDS
 }
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "libavfilter/avfilter.h"
-#include "libavfilter/buffersrc.h"
-#include "libavfilter/buffersink.h"
-#include "libavutil/hwcontext_qsv.h"
-#include "libavutil/hwcontext.h"
-#include "lang.h"
-#include "audio_resample.h"
-
 AVBufferRef *enc_hw_frames_ctx = NULL;
 extern EncQSVFramesContext hb_enc_qsv_frames_ctx;
 AVBufferRef *hb_hw_device_ctx = NULL;
diff --git a/libhb/qsv_filter.c b/libhb/qsv_filter.c
index b330c78ad809f98e2a419dba78350af295fd8315..925e43a1e21c65ad13def15b19dbc837d5b4a1a4 100644
--- a/libhb/qsv_filter.c
+++ b/libhb/qsv_filter.c
@@ -26,14 +26,14 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "qsv_filter.h"
-#include "qsv_libav.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/qsv_filter.h"
+#include "handbrake/qsv_libav.h"
 
 struct hb_filter_private_s
 {
diff --git a/libhb/qsv_filter_pp.c b/libhb/qsv_filter_pp.c
index ddab314fd5762b1ce3984bb75ad6d6b16888c7a9..512b2bd9f15af46d3fe654dd6ea997f891073f9b 100644
--- a/libhb/qsv_filter_pp.c
+++ b/libhb/qsv_filter_pp.c
@@ -26,16 +26,16 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "qsv_filter_pp.h"
-#include "qsv_filter.h"
-#include "qsv_libav.h"
-#include "qsv_memory.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/qsv_filter_pp.h"
+#include "handbrake/qsv_filter.h"
+#include "handbrake/qsv_libav.h"
+#include "handbrake/qsv_memory.h"
 
 
 static int hb_qsv_filter_pre_init( hb_filter_object_t * filter,
diff --git a/libhb/qsv_libav.c b/libhb/qsv_libav.c
index 797aee184a99f33fff0ed396bb78a3c04d6e4dbb..0a6a1a6ce3b0ec38f18ddaf7a69272f72357ac3f 100644
--- a/libhb/qsv_libav.c
+++ b/libhb/qsv_libav.c
@@ -26,12 +26,12 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
-#include "hbffmpeg.h"
-#include "qsv_libav.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/qsv_libav.h"
 
 int hb_qsv_get_free_encode_task(hb_qsv_list * tasks)
 {
diff --git a/libhb/qsv_memory.c b/libhb/qsv_memory.c
index bb6c91a6c019c5cf96b6923d8cbbe3c70fe85a1d..625674910662e9622701297c8632bf3c7cbb15cb 100644
--- a/libhb/qsv_memory.c
+++ b/libhb/qsv_memory.c
@@ -26,13 +26,13 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_QSV
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "qsv_memory.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/qsv_memory.h"
 
 int qsv_nv12_to_yuv420(struct SwsContext* sws_context,hb_buffer_t* dst, mfxFrameSurface1* src, mfxCoreInterface *core){
     int ret = 0;
diff --git a/libhb/reader.c b/libhb/reader.c
index a0496a8f5bac77663ba5d97fd92dc286cbbb6516..2f9f722880e83fc4e80a109451518027acf86e53 100644
--- a/libhb/reader.c
+++ b/libhb/reader.c
@@ -6,7 +6,7 @@
    It may be used under the terms of the GNU General Public License v2.
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 static int  reader_init( hb_work_object_t * w, hb_job_t * job );
 static void reader_close( hb_work_object_t * w );
diff --git a/libhb/rendersub.c b/libhb/rendersub.c
index a909ce4f2870a38fc07be82902fde640a115d405..40613598efc7cad0929cae9fec8c2b87496579c3 100644
--- a/libhb/rendersub.c
+++ b/libhb/rendersub.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 #include <ass/ass.h>
 
 #define ABS(a) ((a) > 0 ? (a) : (-(a)))
diff --git a/libhb/rotate.c b/libhb/rotate.c
index 7ba1405141f3860eb0e9783aec71de7c72be3982..c21c4d8029bea68585bdd3ccda2fcc1f0f9e7b2c 100644
--- a/libhb/rotate.c
+++ b/libhb/rotate.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "common.h"
-#include "avfilter_priv.h"
+#include "handbrake/common.h"
+#include "handbrake/avfilter_priv.h"
 
 static int rotate_init(hb_filter_object_t * filter, hb_filter_init_t * init);
 
diff --git a/libhb/scan.c b/libhb/scan.c
index 97567c8b38d527f34930b0fd623c9786542384d9..9aed75d19592d21244562ecdf1a2c38c5fea212a 100644
--- a/libhb/scan.c
+++ b/libhb/scan.c
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 
 typedef struct
 {
diff --git a/libhb/ssautil.c b/libhb/ssautil.c
index 291620dfe6d1bef859e77eec4baec748c914b4b7..8ff5ea50f72bda202e9fd6a85031257aa0bd55d9 100644
--- a/libhb/ssautil.c
+++ b/libhb/ssautil.c
@@ -8,8 +8,8 @@
  */
 #include <stdio.h>
 #include <ctype.h>
-#include "handbrake.h"
-#include "ssautil.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/ssautil.h"
 
 struct hb_subtitle_style_s
 {
diff --git a/libhb/stream.c b/libhb/stream.c
index aaeaf022e4e71169f5f8ff6d7997fdd5fec94638..feb390c07f5da731b9d61564fe20b02588852214 100644
--- a/libhb/stream.c
+++ b/libhb/stream.c
@@ -11,9 +11,9 @@
 #include <ctype.h>
 #include <errno.h>
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
-#include "lang.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
+#include "handbrake/lang.h"
 #include "libbluray/bluray.h"
 
 #define min(a, b) a < b ? a : b
diff --git a/libhb/sync.c b/libhb/sync.c
index 0af052471ebccfd09592fd9aff563a55fc762028..446129b090f9086e1c6ab06ce314e31280c08502 100644
--- a/libhb/sync.c
+++ b/libhb/sync.c
@@ -7,10 +7,10 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "hbffmpeg.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/hbffmpeg.h"
 #include <stdio.h>
-#include "audio_resample.h"
+#include "handbrake/audio_resample.h"
 
 #define SYNC_MAX_VIDEO_QUEUE_LEN    40
 #define SYNC_MIN_VIDEO_QUEUE_LEN    20
diff --git a/libhb/taskset.c b/libhb/taskset.c
index e829fd79bf3e917dbe33ff112c3dedcf9fe1206d..936ef4de57abb812eaf40d04bd5e790f3366dfd6 100644
--- a/libhb/taskset.c
+++ b/libhb/taskset.c
@@ -7,9 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
-#include "ports.h"
-#include "taskset.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/ports.h"
+#include "handbrake/taskset.h"
 
 int
 taskset_init( taskset_t *ts, int thread_count, size_t arg_size )
diff --git a/libhb/unsharp.c b/libhb/unsharp.c
index f47c6a1bdea56b088c6c45becbd2d508fda42d1e..bf5e8b99b240805eb6e353443cf44192b91ba7ac 100644
--- a/libhb/unsharp.c
+++ b/libhb/unsharp.c
@@ -8,7 +8,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #define UNSHARP_STRENGTH_LUMA_DEFAULT 0.25
 #define UNSHARP_SIZE_LUMA_DEFAULT 7
diff --git a/libhb/vce_common.c b/libhb/vce_common.c
index da0476a642207716c8bf2040e3ec7fdb92646dd9..51d9c07ac2774e95b26a4591a4772422ea716cbd 100644
--- a/libhb/vce_common.c
+++ b/libhb/vce_common.c
@@ -7,13 +7,13 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "project.h"
+#include "handbrake/project.h"
 
 #if HB_PROJECT_FEATURE_VCE
 #include "AMF/core/Factory.h"
 #include "AMF/components/VideoEncoderVCE.h"
 #include "AMF/components/VideoEncoderHEVC.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 AMF_RESULT check_component_available(const wchar_t *componentID)
 {
diff --git a/libhb/vfr.c b/libhb/vfr.c
index a42a5342e7ab17cda82808b3040e5382e0fc6cd5..728b3264422041c04d2ab4db8b14d6dfa506b9a0 100644
--- a/libhb/vfr.c
+++ b/libhb/vfr.c
@@ -7,7 +7,7 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 //#define HB_DEBUG_CFR_DROPS 1
 #define MAX_FRAME_ANALYSIS_DEPTH 10
diff --git a/libhb/work.c b/libhb/work.c
index 7d096aca540cda199f92719310f2bb38b4193e4c..05bd0b04c0661c8d030a6134eb81d770edb0a487 100644
--- a/libhb/work.c
+++ b/libhb/work.c
@@ -7,14 +7,14 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "libavformat/avformat.h"
-#include "decomb.h"
-#include "hbavfilter.h"
+#include "handbrake/decomb.h"
+#include "handbrake/hbavfilter.h"
 
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_common.h"
-#include "qsv_filter_pp.h"
+#include "handbrake/qsv_common.h"
+#include "handbrake/qsv_filter_pp.h"
 #endif
 
 typedef struct
diff --git a/macosx/HBAudio.m b/macosx/HBAudio.m
index 0d825ddf6a188a51834469a68ef90800c3ee6704..046d9a8e7bf54c4a204fddf58d5b0ffac6261a0d 100644
--- a/macosx/HBAudio.m
+++ b/macosx/HBAudio.m
@@ -17,7 +17,7 @@
 #import "HBLocalizationUtilities.h"
 #import "HBJob+Private.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #define NONE_TRACK_INDEX 0
 
diff --git a/macosx/HBAudioDefaults.m b/macosx/HBAudioDefaults.m
index ac05c2b25f9e261ea5dccab62f0857799c311e68..a59b55cefe0a1ecb739c650a01b51556b7a781b7 100644
--- a/macosx/HBAudioDefaults.m
+++ b/macosx/HBAudioDefaults.m
@@ -9,8 +9,8 @@
 #import "HBCodingUtilities.h"
 #import "HBMutablePreset.h"
 
-#import "handbrake.h"
-#import "lang.h"
+#import "handbrake/handbrake.h"
+#import "handbrake/lang.h"
 
 @interface HBAudioDefaults ()
 
diff --git a/macosx/HBAudioTrack.m b/macosx/HBAudioTrack.m
index 0ac7f619d40302754bec9a46625e634afec7620f..1f4b8b03420888218f7d0ab99509266b7753ef52 100644
--- a/macosx/HBAudioTrack.m
+++ b/macosx/HBAudioTrack.m
@@ -9,7 +9,7 @@
 #import "HBJob.h"
 #import "HBCodingUtilities.h"
 #import "HBTitle.h"
-#import "handbrake.h"
+#import "handbrake/handbrake.h"
 
 #define DEFAULT_SAMPLERATE 48000
 
diff --git a/macosx/HBAudioTrackPreset.m b/macosx/HBAudioTrackPreset.m
index ea286c3f4ef7ac568c86f0cc83eb13457994e95b..94542de7edc2b1cbfe9eb38a263a7bb1d68fb6bc 100644
--- a/macosx/HBAudioTrackPreset.m
+++ b/macosx/HBAudioTrackPreset.m
@@ -6,7 +6,7 @@
 
 #import "HBAudioTrackPreset.h"
 #import "HBCodingUtilities.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #define DEFAULT_SAMPLERATE 48000
 
diff --git a/macosx/HBAudioTransformers.m b/macosx/HBAudioTransformers.m
index 95cd661161fbe7757dddd9669720d1ac2bccddfa..07a95001b59ff4949a5727240ae11613c1661efc 100644
--- a/macosx/HBAudioTransformers.m
+++ b/macosx/HBAudioTransformers.m
@@ -6,7 +6,7 @@
 
 #import "HBAudioTransformers.h"
 #import "HBLocalizationUtilities.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 #pragma mark - Value Transformers
 
diff --git a/macosx/HBFilters+UIAdditions.m b/macosx/HBFilters+UIAdditions.m
index d76f25febfb6552a09e44a7c04cb73b342527db2..443d0f884035a68c0604c78db7ac7ac80f31d0e5 100644
--- a/macosx/HBFilters+UIAdditions.m
+++ b/macosx/HBFilters+UIAdditions.m
@@ -7,7 +7,7 @@
 #import "HBFilters+UIAdditions.h"
 #import "HBLocalizationUtilities.h"
 
-#import "handbrake.h"
+#import "handbrake/handbrake.h"
 
 /**
  *  Converts a hb_filter_param_t * array to a NSArray of NSString.
diff --git a/macosx/HBFilters.m b/macosx/HBFilters.m
index d969d82ab215d506d72663f37000a15a3cf3b751..077c85e8d4523c454431c717b97025307aa9698f 100644
--- a/macosx/HBFilters.m
+++ b/macosx/HBFilters.m
@@ -10,7 +10,7 @@
 #import "NSDictionary+HBAdditions.h"
 #import "HBMutablePreset.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 NSString * const HBFiltersChangedNotification = @"HBFiltersChangedNotification";
 
diff --git a/macosx/HBImageUtilities.m b/macosx/HBImageUtilities.m
index 61d1fcfae8e3b09345d991b2e6855716962dea3d..f0c296a785e430ba3f024b3be12a76888438e72b 100644
--- a/macosx/HBImageUtilities.m
+++ b/macosx/HBImageUtilities.m
@@ -6,7 +6,7 @@
 
 #import "HBImageUtilities.h"
 #import <Cocoa/Cocoa.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 CGImageRef CreateScaledCGImageFromCGImage(CGImageRef image, CGFloat thumbnailHeight)
 {
diff --git a/macosx/HBJob+HBAdditions.m b/macosx/HBJob+HBAdditions.m
index 552b8570eb421df2ba527d3f7db52c5d9013d793..3d8a211bc64ecb6576ef0f27bfb57ff143173fc4 100644
--- a/macosx/HBJob+HBAdditions.m
+++ b/macosx/HBJob+HBAdditions.m
@@ -6,7 +6,7 @@
 
 #import "HBJob+HBAdditions.h"
 #import "HBPreferencesKeys.h"
-#import "handbrake.h"
+#import "handbrake/handbrake.h"
 
 static NSDateFormatter *_timeFormatter = nil;
 static NSDateFormatter *_dateFormatter = nil;
diff --git a/macosx/HBJob+HBJobConversion.h b/macosx/HBJob+HBJobConversion.h
index fdbca64e48ae964189a910e1bda3580c6ab709c0..44fe60d600bedf405a9ecd4769ce8f250932a721 100644
--- a/macosx/HBJob+HBJobConversion.h
+++ b/macosx/HBJob+HBJobConversion.h
@@ -6,7 +6,7 @@
 
 #import <Foundation/Foundation.h>
 #import "HBJob.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 @interface HBJob (HBJobConversion)
 
diff --git a/macosx/HBJob+UIAdditions.m b/macosx/HBJob+UIAdditions.m
index 3ce379eed9c83290749eed620f563209ae4bc996..85118083edb5343ae7e1f2c6f508a795ecafccc8 100644
--- a/macosx/HBJob+UIAdditions.m
+++ b/macosx/HBJob+UIAdditions.m
@@ -20,7 +20,7 @@
 
 #import "HBLocalizationUtilities.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 // Text Styles
 static NSDictionary            *detailAttr;
diff --git a/macosx/HBJob.m b/macosx/HBJob.m
index 5960ee51b38ad33a2aa241ce558fc1a4d4f11ec5..8fe439212450fe950786ea1d7a88ae672fbaaedb 100644
--- a/macosx/HBJob.m
+++ b/macosx/HBJob.m
@@ -17,7 +17,7 @@
 #import "HBUtilities.h"
 #import "HBSecurityAccessToken.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 NSString *HBContainerChangedNotification = @"HBContainerChangedNotification";
 NSString *HBChaptersChangedNotification  = @"HBChaptersChangedNotification";
diff --git a/macosx/HBLanguagesSelection.m b/macosx/HBLanguagesSelection.m
index 3fedddd6d691dc65415e116103b4e8aa8d214ddd..018c848752530044176a27b97ee4fe622350505d 100644
--- a/macosx/HBLanguagesSelection.m
+++ b/macosx/HBLanguagesSelection.m
@@ -5,7 +5,7 @@
  It may be used under the terms of the GNU General Public License. */
 
 #import "HBLanguagesSelection.h"
-#include "lang.h"
+#include "handbrake/lang.h"
 
 @implementation HBLang
 
diff --git a/macosx/HBPicture+UIAdditions.m b/macosx/HBPicture+UIAdditions.m
index 9250d0211d0f0717cb202673ddd8211d94696e81..41b40425e4c407d5b1c425ceefa61b315490fe36 100644
--- a/macosx/HBPicture+UIAdditions.m
+++ b/macosx/HBPicture+UIAdditions.m
@@ -8,7 +8,7 @@
 #import "HBTitle.h"
 #import "HBLocalizationUtilities.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 @implementation HBPicture (UIAdditions)
 
diff --git a/macosx/HBPicture.m b/macosx/HBPicture.m
index 6048714d1aab6610e808fd29eae3c490abfc1d29..a623d96001f106ff53cf6a5d68c6bf4d8fb81f1d 100644
--- a/macosx/HBPicture.m
+++ b/macosx/HBPicture.m
@@ -10,7 +10,7 @@
 #import "HBCodingUtilities.h"
 #import "HBMutablePreset.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 NSString * const HBPictureChangedNotification = @"HBPictureChangedNotification";
 
diff --git a/macosx/HBPreset.m b/macosx/HBPreset.m
index 917657c7484eab57c36744cf8431eca6b174f7ba..81b0b51b2cc9f000ebebb685e7f70c1e9a739da7 100644
--- a/macosx/HBPreset.m
+++ b/macosx/HBPreset.m
@@ -7,7 +7,7 @@
 #import "HBPreset.h"
 #import "HBMutablePreset.h"
 
-#include "preset.h"
+#include "handbrake/preset.h"
 
 #import "NSJSONSerialization+HBAdditions.h"
 #import "HBLocalizationUtilities.h"
diff --git a/macosx/HBPresetsManager.m b/macosx/HBPresetsManager.m
index e5b3b3173dabe3d7222302ce9bf27b5abc3afbc7..ac2ea95c12e64f2ce2325e92db6c66d6fed7cfe5 100644
--- a/macosx/HBPresetsManager.m
+++ b/macosx/HBPresetsManager.m
@@ -10,7 +10,7 @@
 #import "HBUtilities.h"
 #import "NSJSONSerialization+HBAdditions.h"
 
-#include "preset.h"
+#include "handbrake/preset.h"
 
 NSString *HBPresetsChangedNotification = @"HBPresetsChangedNotification";
 
diff --git a/macosx/HBStateFormatter+Private.h b/macosx/HBStateFormatter+Private.h
index e7d2e3624461fc670ce0339b3ad12ca59ccd278c..2f4f261a802f93c047a513323255794698f5cf4a 100644
--- a/macosx/HBStateFormatter+Private.h
+++ b/macosx/HBStateFormatter+Private.h
@@ -8,7 +8,7 @@
 
 #import <Foundation/Foundation.h>
 #import "HBStateFormatter.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
diff --git a/macosx/HBStateFormatter.m b/macosx/HBStateFormatter.m
index 39b69a6b2baa7c856bd32d8c515887f51efc6dd0..6d7ca5e16e34add4a11944e462c198054dc67bca 100644
--- a/macosx/HBStateFormatter.m
+++ b/macosx/HBStateFormatter.m
@@ -6,7 +6,7 @@
 
 #import "HBStateFormatter.h"
 #import "HBLocalizationUtilities.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 @implementation HBStateFormatter
 
diff --git a/macosx/HBSubtitles.m b/macosx/HBSubtitles.m
index 8119813eab67b41967db6b5ec4c50be42328d426..9d2833df4461e3ed163be4771647187b17b37f2b 100644
--- a/macosx/HBSubtitles.m
+++ b/macosx/HBSubtitles.m
@@ -18,7 +18,7 @@
 #import "HBJob+Private.h"
 #import "HBSecurityAccessToken.h"
 
-#include "common.h"
+#include "handbrake/common.h"
 
 #define NONE_TRACK_INDEX        0
 #define FOREIGN_TRACK_INDEX     1
diff --git a/macosx/HBSubtitlesTrack.m b/macosx/HBSubtitlesTrack.m
index 019c89cf90f6969b1a6333220a0243dd4e09513b..457410a270b71dc79628f713f5562d269bab064a 100644
--- a/macosx/HBSubtitlesTrack.m
+++ b/macosx/HBSubtitlesTrack.m
@@ -8,8 +8,8 @@
 #import "HBCodingUtilities.h"
 #import "HBTitle.h"
 
-#include "common.h"
-#include "lang.h"
+#include "handbrake/common.h"
+#include "handbrake/lang.h"
 
 #define CHAR_CODE_DEFAULT_INDEX 28
 
diff --git a/macosx/HBTitle+Private.h b/macosx/HBTitle+Private.h
index 7afb7defc0d97629aef7c02c80b6ba6a153b7b73..57e4cb9e661b206544e562759efe93b5cfeadba9 100644
--- a/macosx/HBTitle+Private.h
+++ b/macosx/HBTitle+Private.h
@@ -7,7 +7,7 @@
 #import <Foundation/Foundation.h>
 #import "HBTitle.h"
 #import "HBPreset.h"
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 @interface HBTitle (Private)
 
diff --git a/macosx/HBTitle.m b/macosx/HBTitle.m
index dbb52a080f9ea4357c705076605b64c1539d0d2a..895ef6fe8b044b6ece79dd098d59ec0316a95841 100644
--- a/macosx/HBTitle.m
+++ b/macosx/HBTitle.m
@@ -14,7 +14,7 @@
 #import "HBSecurityAccessToken.h"
 #import "HBUtilities.h"
 
-#include "lang.h"
+#include "handbrake/lang.h"
 
 @interface HBMetadata ()
 
diff --git a/macosx/HBUtilities.m b/macosx/HBUtilities.m
index 1f2810a52207b725859f631e52d755c244109dbf..4015dd916c8cb03d595552dd0b8a4d088efd41cf 100644
--- a/macosx/HBUtilities.m
+++ b/macosx/HBUtilities.m
@@ -7,7 +7,7 @@
 #import "HBUtilities.h"
 #import <Cocoa/Cocoa.h>
 
-#include "lang.h"
+#include "handbrake/lang.h"
 
 static BOOL hb_resolveBookmarks = YES;
 
diff --git a/macosx/HBVideo+UIAdditions.m b/macosx/HBVideo+UIAdditions.m
index a2c1b4d3930f5e5f734b2f393d9f1de9cbbda2a5..cd9823698c6819f1cb6e2e917187fbd43eef015e 100644
--- a/macosx/HBVideo+UIAdditions.m
+++ b/macosx/HBVideo+UIAdditions.m
@@ -8,7 +8,7 @@
 #import "HBJob+Private.h"
 #import "HBLocalizationUtilities.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 @implementation HBVideo (UIAdditions)
 
diff --git a/macosx/HBVideo.m b/macosx/HBVideo.m
index 815292157e90b12d49565fd1680a90c3c3852060..c062759e0c5aa53711d3766d396d9db32a6cef1e 100644
--- a/macosx/HBVideo.m
+++ b/macosx/HBVideo.m
@@ -9,7 +9,7 @@
 #import "HBCodingUtilities.h"
 #import "HBMutablePreset.h"
 
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 
 NSString * const HBVideoChangedNotification = @"HBVideoChangedNotification";
 
diff --git a/macosx/HandBrake.xcodeproj/project.pbxproj b/macosx/HandBrake.xcodeproj/project.pbxproj
index f96e2958f0c20689084a872e3629e4a06d1ef12e..2b30ed702e04831cbddfa7bb2069f2728e3377f6 100644
--- a/macosx/HandBrake.xcodeproj/project.pbxproj
+++ b/macosx/HandBrake.xcodeproj/project.pbxproj
@@ -1860,8 +1860,8 @@
 			outputFileListPaths = (
 			);
 			outputPaths = (
-				"$(EXTERNAL_BUILD)/libhb/hb_dict.h",
-				"$(EXTERNAL_BUILD)/libhb/handbrake.h",
+				"$(EXTERNAL_BUILD)/libhb/handbrake/hb_dict.h",
+				"$(EXTERNAL_BUILD)/libhb/handbrake/handbrake.h",
 				"$(EXTERNAL_BUILD)/macosx/Info.plist",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
diff --git a/macosx/NSDictionary+HBAdditions.h b/macosx/NSDictionary+HBAdditions.h
index b653d1b5984624fb6dc920051413c3c635e77d4d..42fc79a5438866c14153f933964db61ecc57a532 100644
--- a/macosx/NSDictionary+HBAdditions.h
+++ b/macosx/NSDictionary+HBAdditions.h
@@ -5,7 +5,7 @@
  It may be used under the terms of the GNU General Public License. */
 
 #import <Foundation/Foundation.h>
-#include "hb_dict.h"
+#include "handbrake/hb_dict.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
diff --git a/test/parsecsv.c b/test/parsecsv.c
index b777d1dab7de100cd98763217b933f047a8455a7..95a6f883ae8315e6db2340f72d091881403db6f0 100644
--- a/test/parsecsv.c
+++ b/test/parsecsv.c
@@ -8,7 +8,7 @@
  */
 
 #include <fcntl.h>
-#include "handbrake.h"
+#include "handbrake/handbrake.h"
 #include "parsecsv.h"
 
 /* Internal declarations */
diff --git a/test/test.c b/test/test.c
index c4abe76872445cdc4f6f0652ae82e3c895ff027c..227340fb6114036902d85f249d33f11f84c1e0be 100644
--- a/test/test.c
+++ b/test/test.c
@@ -30,12 +30,12 @@
 #include <pthread.h>
 #endif
 
-#include "handbrake.h"
-#include "lang.h"
+#include "handbrake/handbrake.h"
+#include "handbrake/lang.h"
 #include "parsecsv.h"
 
 #if HB_PROJECT_FEATURE_QSV
-#include "qsv_common.h"
+#include "handbrake/qsv_common.h"
 #endif
 
 #if defined( __APPLE_CC__ )
