diff --git a/libhb/handbrake/audio_remap.h b/libhb/handbrake/audio_remap.h
index d8f1ed3b6952210741de9341781b5dbe66f616c8..aca8b301d6374e7b7beef2610ce3eb5d11d807cb 100644
--- a/libhb/handbrake/audio_remap.h
+++ b/libhb/handbrake/audio_remap.h
@@ -23,8 +23,8 @@
  * Left/Right Surround == Side Left/Right
  * Left/Right Rear Surround == Back Left/Right */
 
-#ifndef AUDIO_REMAP_H
-#define AUDIO_REMAP_H
+#ifndef HANDBRAKE_AUDIO_REMAP_H
+#define HANDBRAKE_AUDIO_REMAP_H
 
 #include <stdint.h>
 #include "libavutil/samplefmt.h"
@@ -101,4 +101,4 @@ void              hb_audio_remap_build_table(hb_chan_map_t *channel_map_out,
                                              uint64_t channel_layout,
                                              int *remap_table);
 
-#endif /* AUDIO_REMAP_H */
+#endif /* HANDBRAKE_AUDIO_REMAP_H */
diff --git a/libhb/handbrake/audio_resample.h b/libhb/handbrake/audio_resample.h
index 1032377581b509d33fa87547397a145e60c79208..32a724d5a3ae7053a985f866f196d53bc5f27395 100644
--- a/libhb/handbrake/audio_resample.h
+++ b/libhb/handbrake/audio_resample.h
@@ -14,8 +14,8 @@
  * sample_rate conversion will come later (libswresample doesn't support
  * sample_rate conversion with float samples yet). */
 
-#ifndef AUDIO_RESAMPLE_H
-#define AUDIO_RESAMPLE_H
+#ifndef HANDBRAKE_AUDIO_RESAMPLE_H
+#define HANDBRAKE_AUDIO_RESAMPLE_H
 
 #include <math.h>
 #include <stdint.h>
@@ -117,4 +117,4 @@ void                 hb_audio_resample_free(hb_audio_resample_t *resample);
 hb_buffer_t*         hb_audio_resample(hb_audio_resample_t *resample,
                                        const uint8_t **samples, int nsamples);
 
-#endif /* AUDIO_RESAMPLE_H */
+#endif /* HANDBRAKE_AUDIO_RESAMPLE_H */
diff --git a/libhb/handbrake/avfilter_priv.h b/libhb/handbrake/avfilter_priv.h
index 20e616cc236b9b837c6da2e1a290f54900a25067..629a042286e5680818a3a0fa6605d88bdd69cb5d 100644
--- a/libhb/handbrake/avfilter_priv.h
+++ b/libhb/handbrake/avfilter_priv.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_AVFILTER_PRIV_H
-#define HB_AVFILTER_PRIV_H
+#ifndef HANDBRAKE_AVFILTER_PRIV_H
+#define HANDBRAKE_AVFILTER_PRIV_H
 
 #include "libavfilter/avfilter.h"
 #include "handbrake/hbavfilter.h"
@@ -31,4 +31,4 @@ int  hb_avfilter_null_work( hb_filter_object_t * filter,
                             hb_buffer_t ** buf_in, hb_buffer_t ** buf_out );
 void hb_avfilter_alias_close( hb_filter_object_t * filter );
 
-#endif // HB_AVFILTER_PRIV_H
+#endif // HANDBRAKE_AVFILTER_PRIV_H
diff --git a/libhb/handbrake/bits.h b/libhb/handbrake/bits.h
index 8785f4b5caf0a3b16763d2cc9d3a3b0bf34e4702..5a069a840534a9dd598b4b5eb6771ec760443b2b 100644
--- a/libhb/handbrake/bits.h
+++ b/libhb/handbrake/bits.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_BITS_H
-#define HB_BITS_H
+#ifndef HANDBRAKE_BITS_H
+#define HANDBRAKE_BITS_H
 
 static inline int
 allbits_set(uint32_t *bitmap, int num_words)
@@ -87,4 +87,4 @@ bit_nset(uint32_t *bit_map, int start_pos, int stop_pos)
     }
 }
 
-#endif /* HB_BITS_H */
+#endif /* HANDBRAKE_BITS_H */
diff --git a/libhb/handbrake/colormap.h b/libhb/handbrake/colormap.h
index 7ea9ac57c0e6684957eb15bc056ad257aa5eee22..3486a80ff99e446b9d49fba8f640af1c9ff79805 100644
--- a/libhb/handbrake/colormap.h
+++ b/libhb/handbrake/colormap.h
@@ -7,8 +7,8 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_COLORMAP_H
-#define HB_COLORMAP_H
+#ifndef HANDBRAKE_COLORMAP_H
+#define HANDBRAKE_COLORMAP_H
 
 #define HB_RGB_TO_BGR(c)    (((c & 0xff0000) >> 16) | \
                              ((c & 0x00ff00)      ) | \
@@ -17,4 +17,4 @@
 
 uint32_t hb_rgb_lookup_by_name(const char *color);
 
-#endif // HB_COLORMAP_H
+#endif // HANDBRAKE_COLORMAP_H
diff --git a/libhb/handbrake/common.h b/libhb/handbrake/common.h
index 8ab21fa8abc6933f58aa221d0fcbbd14b53f0599..2f4826d242fe0bfaf9bb1fdb7a41759bae43eb4d 100644
--- a/libhb/handbrake/common.h
+++ b/libhb/handbrake/common.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
-#ifndef HB_COMMON_H
-#define HB_COMMON_H
+#ifndef HANDBRAKE_COMMON_H
+#define HANDBRAKE_COMMON_H
 
 #include "handbrake/project.h"
 #include "handbrake/hbtypes.h"
@@ -1443,4 +1443,4 @@ int hb_output_color_matrix(hb_job_t * job);
 #define HB_BOOL_REG      "(yes|no|true|false|[01])"
 #define HB_ALL_REG       "(.*)"
 
-#endif
+#endif // HANDBRAKE_COMMON_H
diff --git a/libhb/handbrake/compat.h b/libhb/handbrake/compat.h
index f8ab40614e1f01d442e5e2ef3635b1b2335b8e8b..1ddd96c8646fd1ebc0963a201f68a9931fbcfeaa 100644
--- a/libhb/handbrake/compat.h
+++ b/libhb/handbrake/compat.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_COMPAT_H
-#define HB_COMPAT_H
+#ifndef HANDBRAKE_COMPAT_H
+#define HANDBRAKE_COMPAT_H
 
 #ifdef HB_NEED_STRTOK_R
 /*
@@ -35,4 +35,4 @@ int strerror_r(int errnum, char *strerrbuf, size_t buflen);
 #endif // _GNU_SOURCE
 #endif // HAVE_STRERROR_R
 
-#endif // HB_COMPAT_H
+#endif // HANDBRAKE_COMPAT_H
diff --git a/libhb/handbrake/deccc608sub.h b/libhb/handbrake/deccc608sub.h
index 044fc78f763aa9f95c65a43972a26b8303619807..b814dbacc93da732899d26d655d4fb19aa3a007a 100644
--- a/libhb/handbrake/deccc608sub.h
+++ b/libhb/handbrake/deccc608sub.h
@@ -10,8 +10,8 @@
 /*
  * From ccextractor...
  */
-#ifndef __DECCC608SUB_H__
-#define __DECCC608SUB_H__
+#ifndef HANDBRAKE_DECCC608SUB_H
+#define HANDBRAKE_DECCC608SUB_H
 
 #include "handbrake/common.h"
 
@@ -143,4 +143,4 @@ enum output_format
     OF_RCWT = 4
 };
 
-#endif // __DECCC608SUB_H__
+#endif // HANDBRAKE_DECCC608SUB_H
diff --git a/libhb/handbrake/decomb.h b/libhb/handbrake/decomb.h
index 181bb8ad38e0bf929661085955a7cd2efa165e84..bec6ee12e57ea1bd30c11b496493708eb49ed952 100644
--- a/libhb/handbrake/decomb.h
+++ b/libhb/handbrake/decomb.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
-#ifndef HB_DECOMB_H
-#define HB_DECOMB_H
+#ifndef HANDBRAKE_DECOMB_H
+#define HANDBRAKE_DECOMB_H
 
 #define MODE_DECOMB_YADIF       1 // Use yadif
 #define MODE_DECOMB_BLEND       2 // Use blending interpolation
@@ -22,4 +22,4 @@
 #define MODE_YADIF_BOB          4
 #define MODE_DEINTERLACE_QSV    8
 
-#endif // HB_DECOMB_H
+#endif // HANDBRAKE_DECOMB_H
diff --git a/libhb/handbrake/decsrtsub.h b/libhb/handbrake/decsrtsub.h
index d38357d25f8f22e45e0260eaf377d5e0ccb825e7..d81760d32150a5a9b595a241d7113ba3b73f1739 100644
--- a/libhb/handbrake/decsrtsub.h
+++ b/libhb/handbrake/decsrtsub.h
@@ -7,10 +7,10 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef __DECSRTSUB_H__
-#define __DECSRTSUB_H__
+#ifndef HANDBRAKE_DECSRTSUB_H
+#define HANDBRAKE_DECSRTSUB_H
 
 void hb_srt_to_ssa(hb_buffer_t *sub_in, int line);
 
-#endif // __DECSRTSUB_H__
+#endif // HANDBRAKE_DECSRTSUB_H
 
diff --git a/libhb/handbrake/decssasub.h b/libhb/handbrake/decssasub.h
index 19c7ce6332d5523b5da4c3bd7c3238efc4652294..cc05845e0dae2633cc4a2c60e2d9c82f1859057e 100644
--- a/libhb/handbrake/decssasub.h
+++ b/libhb/handbrake/decssasub.h
@@ -7,7 +7,7 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef __DECSSASUB_H__
-#define __DECSSASUB_H__
+#ifndef HANDBRAKE_DECSSASUB_H
+#define HANDBRAKE_DECSSASUB_H
 
-#endif // __DECSSASUB_H__
+#endif // HANDBRAKE_DECSSASUB_H
diff --git a/libhb/handbrake/dvd.h b/libhb/handbrake/dvd.h
index 7ac673ca096acbca490e300398abb9433398fd0b..f26953996eaec0123a9ecab572d6f18886bb75fd 100644
--- a/libhb/handbrake/dvd.h
+++ b/libhb/handbrake/dvd.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_DVD_H
-#define HB_DVD_H
+#ifndef HANDBRAKE_DVD_H
+#define HANDBRAKE_DVD_H
 
 #include "dvdnav/dvdnav.h"
 #include "dvdread/ifo_read.h"
@@ -115,6 +115,4 @@ typedef struct hb_dvd_func_s hb_dvd_func_t;
 hb_dvd_func_t * hb_dvdnav_methods( void );
 hb_dvd_func_t * hb_dvdread_methods( void );
 
-#endif // HB_DVD_H
-
-
+#endif // HANDBRAKE_DVD_H
diff --git a/libhb/handbrake/eedi2.h b/libhb/handbrake/eedi2.h
index e524092f0a5031e29c8f9d94e18baf03e4e90b22..734bad8e87c68d39a7ab0a18c6a9bf04a71a023d 100644
--- a/libhb/handbrake/eedi2.h
+++ b/libhb/handbrake/eedi2.h
@@ -7,6 +7,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
+#ifndef HANDBRAKE_EEDI2_H
+#define HANDBRAKE_EEDI2_H
+
 // Used to order a sequence of metrics for median filtering
 void eedi2_sort_metrics( int *order, const int length );
 
@@ -91,3 +94,5 @@ void eedi2_calc_derivatives( uint8_t *srcp, int src_pitch, int height, int width
 
 void eedi2_post_process_corner( int *x2, int *y2, int *xy, const int pitch, uint8_t * mskp, int msk_pitch,
                                 uint8_t * dstp, int dst_pitch, int height, int width, int field );
+
+#endif // HANDBRAKE_EEDI2_H
diff --git a/libhb/handbrake/encx264.h b/libhb/handbrake/encx264.h
index 29085e50cf7fb5c5e8f12d02e56105466973f852..97bac979a6285687424e0dc56c13f2599b683314 100644
--- a/libhb/handbrake/encx264.h
+++ b/libhb/handbrake/encx264.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_ENCX264_H
-#define HB_ENCX264_H
+#ifndef HANDBRAKE_ENCX264_H
+#define HANDBRAKE_ENCX264_H
 
 #include "x264.h"
 #include "handbrake/h264_common.h"
@@ -62,4 +62,4 @@ typedef struct x264_api_s
 void               hb_x264_global_init(void);
 const x264_api_t * hb_x264_api_get(int bit_depth);
 
-#endif // HB_ENCX264_H
+#endif // HANDBRAKE_ENCX264_H
diff --git a/libhb/handbrake/h264_common.h b/libhb/handbrake/h264_common.h
index afe2401131ed3fcb5ad3eb963b459cbf24a5d083..3f1edc3366798aebfc503286cde750056c50e60d 100644
--- a/libhb/handbrake/h264_common.h
+++ b/libhb/handbrake/h264_common.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_H264_COMMON_H
-#define HB_H264_COMMON_H
+#ifndef HANDBRAKE_H264_COMMON_H
+#define HANDBRAKE_H264_COMMON_H
 
 static const char * const hb_h264_profile_names_8bit[]  = {
     "auto", "high", "main", "baseline", NULL, };
@@ -37,4 +37,4 @@ static const char * const hb_h264_colmatrix_names[] = {
     "GBR", "bt709", "undef", "", "fcc", "bt470bg", "smpte170m",
     "smpte240m", "YCgCo", "bt2020nc", "bt2020c", NULL, };
 
-#endif  //HB_H264_COMMON_H
+#endif // HANDBRAKE_H264_COMMON_H
diff --git a/libhb/handbrake/h265_common.h b/libhb/handbrake/h265_common.h
index 8b5c1072dece6415a8f39b1fe16c31ec18b64528..98fac0ddcbd8b8e4b8bf563afe78c933a559a15e 100644
--- a/libhb/handbrake/h265_common.h
+++ b/libhb/handbrake/h265_common.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_H265_COMMON_H
-#define HB_H265_COMMON_H
+#ifndef HANDBRAKE_H265_COMMON_H
+#define HANDBRAKE_H265_COMMON_H
 
 #include "handbrake/project.h"
 
@@ -53,4 +53,4 @@ static const char * const hb_h265_colmatrix_names[] = {
     "GBR", "bt709", "undef", "", "fcc", "bt470bg", "smpte170m",
     "smpte240m", "YCgCo", "bt2020nc", "bt2020c", NULL, };
 
-#endif  //HB_H265_COMMON_H
+#endif // HANDBRAKE_H265_COMMON_H
diff --git a/libhb/handbrake/handbrake.h b/libhb/handbrake/handbrake.h
index 6143316d56cafe2e1c9960c04cca21f94d50dc82..d9a6f363681f35fe2e2df8d711a314ec7ea426eb 100644
--- a/libhb/handbrake/handbrake.h
+++ b/libhb/handbrake/handbrake.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
  
-#ifndef HB_HB_H
-#define HB_HB_H
+#ifndef HANDBRAKE_HANDBRAKE_H
+#define HANDBRAKE_HANDBRAKE_H
 
 #ifdef __cplusplus
 extern "C" {
@@ -142,4 +142,4 @@ int is_hardware_disabled(void);
 }
 #endif
 
-#endif
+#endif // HANDBRAKE_HANDBRAKE_H
diff --git a/libhb/handbrake/hb_dict.h b/libhb/handbrake/hb_dict.h
index 5d8a188e02ee2039eb720fb8c8aa27af0ec4c19e..cc539dc4c034f39d6b737341b024d9253493aec8 100644
--- a/libhb/handbrake/hb_dict.h
+++ b/libhb/handbrake/hb_dict.h
@@ -6,8 +6,8 @@
    It may be used under the terms of the GNU General Public License v2.
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
-#if !defined(HB_DICT_H)
-#define HB_DICT_H
+#if !defined(HANDBRAKE_DICT_H)
+#define HANDBRAKE_DICT_H
 
 #include "handbrake/hbtypes.h"
 #include <jansson.h>
@@ -166,4 +166,4 @@ char      * hb_dict_to_encopts(const hb_dict_t * dict);
 #define hb_dict_set_bool(dict, key, val) hb_dict_set(dict, key, hb_value_bool(val))
 
 
-#endif // !defined(HB_DICT_H)
+#endif // !defined(HANDBRAKE_DICT_H)
diff --git a/libhb/handbrake/hb_json.h b/libhb/handbrake/hb_json.h
index aac3223f75e5a7b31c6b63f4d594dc167545b842..725072ded5ddfb493d29ee665e480a947847bf88 100644
--- a/libhb/handbrake/hb_json.h
+++ b/libhb/handbrake/hb_json.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_JSON_H
-#define HB_JSON_H
+#ifndef HANDBRAKE_JSON_H
+#define HANDBRAKE_JSON_H
 
 #ifdef __cplusplus
 extern "C" {
@@ -43,4 +43,4 @@ hb_dict_t  * hb_version_dict(void);
 }
 #endif
 
-#endif // HB_JSON_H
+#endif // HANDBRAKE_JSON_H
diff --git a/libhb/handbrake/hbavfilter.h b/libhb/handbrake/hbavfilter.h
index 319b4887142a6e9abed728f9ec803bc5d50c6066..5a75a88b66d9ce544bd33cd2822b68c83e91a973 100644
--- a/libhb/handbrake/hbavfilter.h
+++ b/libhb/handbrake/hbavfilter.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_AVFILTER_H
-#define HB_AVFILTER_H
+#ifndef HANDBRAKE_AVFILTER_H
+#define HANDBRAKE_AVFILTER_H
 
 #include "handbrake/common.h"
 
@@ -39,4 +39,4 @@ void    hb_avfilter_append_dict(hb_value_array_t * filters,
 
 void    hb_avfilter_combine(hb_list_t * list);
 
-#endif // HB_AVFILTER_H
+#endif // HANDBRAKE_AVFILTER_H
diff --git a/libhb/handbrake/hbffmpeg.h b/libhb/handbrake/hbffmpeg.h
index 6357dcd9c2c9796e393feeb4b032238741a2ef37..8612f12d7a19b52fa660f94de476508b767aa4a7 100644
--- a/libhb/handbrake/hbffmpeg.h
+++ b/libhb/handbrake/hbffmpeg.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_FFMPEG_H
-#define HB_FFMPEG_H
+#ifndef HANDBRAKE_FFMPEG_H
+#define HANDBRAKE_FFMPEG_H
 
 #include "libavcodec/avcodec.h"
 #include "libavformat/avformat.h"
@@ -59,4 +59,4 @@ int hb_av_encoder_present(int encoder);
 const char* const* hb_av_profile_get_names(int encoder);
 const char* const* hb_av_level_get_names(int encoder);
 
-#endif
+#endif // HANDBRAKE_FFMPEG_H
diff --git a/libhb/handbrake/hbtypes.h b/libhb/handbrake/hbtypes.h
index 127f9d50cc32d3e7d6908f0d37dff1fcfae44058..403ada0e219aa2fff557748a264d2a0118f1491e 100644
--- a/libhb/handbrake/hbtypes.h
+++ b/libhb/handbrake/hbtypes.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_TYPES_H
-#define HB_TYPES_H
+#ifndef HANDBRAKE_TYPES_H
+#define HANDBRAKE_TYPES_H
 
 typedef struct hb_handle_s hb_handle_t;
 typedef struct hb_list_s hb_list_t;
@@ -45,4 +45,4 @@ typedef struct hb_image_format_s hb_image_format_t;
 typedef struct hb_fifo_s hb_fifo_t;
 typedef struct hb_lock_s hb_lock_t;
 
-#endif // HB_TYPES_H
+#endif // HANDBRAKE_TYPES_H
diff --git a/libhb/handbrake/internal.h b/libhb/handbrake/internal.h
index 93ae27dca74e72834fc45dadee21011ce73ca69a..db17d67d2ac10030f46a818abd5bb4f51489734c 100644
--- a/libhb/handbrake/internal.h
+++ b/libhb/handbrake/internal.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_INTERNAL_H
-#define HB_INTERNAL_H
+#ifndef HANDBRAKE_INTERNAL_H
+#define HANDBRAKE_INTERNAL_H
 
 #include "handbrake/project.h"
 #include "handbrake/hbffmpeg.h"
@@ -538,4 +538,4 @@ void                 hb_chapter_dequeue(hb_chapter_queue_t *q, hb_buffer_t *b);
 #define HB_FONT_SANS "sans-serif"
 #endif
 
-#endif // HB_INTERNAL_H
+#endif // HANDBRAKE_INTERNAL_H
diff --git a/libhb/handbrake/lang.h b/libhb/handbrake/lang.h
index 9017de06688fe98984a6b0f1a4d443992500ffb7..d5bdec3b43c2ce2ca6c036bf5e15f93735f32b3e 100644
--- a/libhb/handbrake/lang.h
+++ b/libhb/handbrake/lang.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_LANG_H
-#define HB_LANG_H
+#ifndef HANDBRAKE_LANG_H
+#define HANDBRAKE_LANG_H
 
 typedef struct iso639_lang_t
 {
@@ -62,4 +62,4 @@ const iso639_lang_t* lang_get_next(const iso639_lang_t *last);
 #ifdef __cplusplus
 }
 #endif
-#endif
+#endif // HANDBRAKE_LANG_H
diff --git a/libhb/handbrake/nal_units.h b/libhb/handbrake/nal_units.h
index 0efccc4763cf382a508a838c317d42005fbce600..00b89fd50f4797c676cf0e9f3d1b05882436a60f 100644
--- a/libhb/handbrake/nal_units.h
+++ b/libhb/handbrake/nal_units.h
@@ -7,8 +7,8 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_NAL_UNITS_H
-#define HB_NAL_UNITS_H
+#ifndef HANDBRAKE_NAL_UNITS_H
+#define HANDBRAKE_NAL_UNITS_H
 
 #include <stdint.h>
 
@@ -48,4 +48,4 @@ uint8_t* hb_annexb_find_next_nalu(const uint8_t *start, size_t *size);
 hb_buffer_t* hb_nal_bitstream_annexb_to_mp4(const uint8_t *data, const size_t size);
 hb_buffer_t* hb_nal_bitstream_mp4_to_annexb(const uint8_t *data, const size_t size, const uint8_t nal_length_size);
 
-#endif // HB_NAL_UNITS_H
+#endif // HANDBRAKE_NAL_UNITS_H
diff --git a/libhb/handbrake/nlmeans.h b/libhb/handbrake/nlmeans.h
index 3ed4d7461e5edc4e21b2808599cd1db951a79374..8c2dec0b94504e28e5b8b5022e78aee0e8204e3a 100644
--- a/libhb/handbrake/nlmeans.h
+++ b/libhb/handbrake/nlmeans.h
@@ -8,6 +8,9 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
+#ifndef HANDBRAKE_NLMEANS_H
+#define HANDBRAKE_NLMEANS_H
+
 typedef struct
 {
     void (*build_integral)(uint32_t *integral,
@@ -25,3 +28,5 @@ typedef struct
 } NLMeansFunctions;
 
 void nlmeans_init_x86(NLMeansFunctions *functions);
+
+#endif // HANDBRAKE_NLMEANS_H
diff --git a/libhb/handbrake/nvenc_common.h b/libhb/handbrake/nvenc_common.h
index f7873b882689a0f94ffdd75352b54f6f8cca77ab..1a94c77fc37fada47283ab8594352a3ccbe6862c 100644
--- a/libhb/handbrake/nvenc_common.h
+++ b/libhb/handbrake/nvenc_common.h
@@ -7,5 +7,10 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
+#ifndef HANDBRAKE_NVENC_COMMON_H
+#define HANDBRAKE_NVENC_COMMON_H
+
 int            hb_nvenc_h264_available();
 int            hb_nvenc_h265_available();
+
+#endif // HANDBRAKE_NVENC_COMMON_H
diff --git a/libhb/handbrake/param.h b/libhb/handbrake/param.h
index 378c5e5788a62e1aca7cfe93874a70e532c44029..a7394346d7b87dbb5daed34c6b6e9b21703f3573 100644
--- a/libhb/handbrake/param.h
+++ b/libhb/handbrake/param.h
@@ -6,8 +6,9 @@
    It may be used under the terms of the GNU General Public License v2.
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
-#ifndef HB_PARAM_H
-#define HB_PARAM_H
+
+#ifndef HANDBRAKE_PARAM_H
+#define HANDBRAKE_PARAM_H
 
 typedef struct hb_filter_param_s hb_filter_param_t;
 
@@ -44,4 +45,4 @@ char ** hb_filter_get_tunes_name(int filter_id);
 char  * hb_filter_get_presets_json(int filter_id);
 char  * hb_filter_get_tunes_json(int filter_id);
 
-#endif // HB_PARAM_H
+#endif // HANDBRAKE_PARAM_H
diff --git a/libhb/handbrake/plist.h b/libhb/handbrake/plist.h
index 12b51dd2e3bddcc9df45a7800e6a6ec11af3ee02..0cbf44ddf903e5a8f54db068346194513e997de6 100644
--- a/libhb/handbrake/plist.h
+++ b/libhb/handbrake/plist.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#if !defined(_HB_PLIST_H_)
-#define _HB_PLIST_H_
+#if !defined(HANDBRAKE_PLIST_H)
+#define HANDBRAKE_PLIST_H
 
 #include <stdio.h>
 #include "handbrake/hb_dict.h"
@@ -18,5 +18,4 @@ hb_value_t * hb_plist_parse_file(const char *filename);
 void         hb_plist_write(FILE *file, hb_value_t  *val);
 void         hb_plist_write_file(const char *filename, hb_value_t  *val);
 
-#endif // _HB_PLIST_H_
-
+#endif // HANDBRAKE_PLIST_H
diff --git a/libhb/handbrake/ports.h b/libhb/handbrake/ports.h
index 6b793da28210c69a9ff30f4eebebbc4821501fc4..788d474df9df186e4ad93eb1f5aadd026957a78f 100644
--- a/libhb/handbrake/ports.h
+++ b/libhb/handbrake/ports.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_PORTS_H
-#define HB_PORTS_H
+#ifndef HANDBRAKE_PORTS_H
+#define HANDBRAKE_PORTS_H
 
 #if ARCH_X86_64 || ARCH_X86_32
 #define ARCH_X86
@@ -236,5 +236,4 @@ int    hb_dlclose(void *h);
 
 #endif /* __LIBHB__ */
 
-#endif
-
+#endif // HANDBRAKE_PORTS_H
diff --git a/libhb/handbrake/preset.h b/libhb/handbrake/preset.h
index abb188487b8f3574b920a50b295b1d579051a965..0c8f4b7176a861d385968408bb8ec57a0957b478 100644
--- a/libhb/handbrake/preset.h
+++ b/libhb/handbrake/preset.h
@@ -6,8 +6,8 @@
    It may be used under the terms of the GNU General Public License v2.
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
-#if !defined(HB_PRESET_H)
-#define HB_PRESET_H
+#if !defined(HANDBRAKE_PRESET_H)
+#define HANDBRAKE_PRESET_H
 
 #include "handbrake/common.h"
 #include "handbrake/hb_dict.h"
@@ -200,4 +200,4 @@ char       * hb_presets_package_json(const hb_value_t *presets);
 // and return as json string
 char       * hb_presets_json_package(const char *json_presets);
 
-#endif // HB_PRESET_H
+#endif // HANDBRAKE_PRESET_H
diff --git a/libhb/handbrake/qsv_common.h b/libhb/handbrake/qsv_common.h
index ca844c5e9e983fc331a4ba469aced6adccaef9d4..9f73d97fb8c9c61acc3184b3276d176e8f7bae13 100644
--- a/libhb/handbrake/qsv_common.h
+++ b/libhb/handbrake/qsv_common.h
@@ -7,8 +7,8 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_QSV_COMMON_H
-#define HB_QSV_COMMON_H
+#ifndef HANDBRAKE_QSV_COMMON_H
+#define HANDBRAKE_QSV_COMMON_H
 
 int            hb_qsv_available();
 
@@ -247,4 +247,4 @@ void hb_qsv_uninit_dec(AVCodecContext *s);
 void hb_qsv_uninit_enc();
 
 #endif // HB_PROJECT_FEATURE_QSV
-#endif // HB_QSV_COMMON_H
+#endif // HANDBRAKE_QSV_COMMON_H
diff --git a/libhb/handbrake/qsv_filter.h b/libhb/handbrake/qsv_filter.h
index 6d2650059b2ef16cee971ad4e0366377f1faa86f..3c1cf1259e46134a98346c7fc6f0f151b33a37ab 100644
--- a/libhb/handbrake/qsv_filter.h
+++ b/libhb/handbrake/qsv_filter.h
@@ -26,8 +26,8 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#ifndef QSV_FILTER_H
-#define QSV_FILTER_H
+#ifndef HANDBRAKE_QSV_FILTER_H
+#define HANDBRAKE_QSV_FILTER_H
 
 #include "handbrake/project.h"
 
@@ -35,4 +35,4 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 void qsv_filter_close( hb_qsv_context* qsv, HB_QSV_STAGE_TYPE vpp_type );
 #endif
 
-#endif // QSV_FILTER_H
+#endif // HANDBRAKE_QSV_FILTER_H
diff --git a/libhb/handbrake/qsv_filter_pp.h b/libhb/handbrake/qsv_filter_pp.h
index c0e7f605e5623cd467a0d9c342e17f9baf8d328a..85c2b7becb4f87fad8be2a01395a61ea0e419c12 100644
--- a/libhb/handbrake/qsv_filter_pp.h
+++ b/libhb/handbrake/qsv_filter_pp.h
@@ -26,8 +26,8 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#ifndef QSV_FILTER_PP_H
-#define QSV_FILTER_PP_H
+#ifndef HANDBRAKE_QSV_FILTER_PP_H
+#define HANDBRAKE_QSV_FILTER_PP_H
 
 #include "handbrake/project.h"
 
@@ -114,4 +114,4 @@ mfxStatus unlock_frame(mfxFrameAllocator *,mfxFrameSurface1*);
 
 
 #endif // HB_PROJECT_FEATURE_QSV
-#endif //QSV_FILTER_PP_H
+#endif // HANDBRAKE_QSV_FILTER_PP_H
diff --git a/libhb/handbrake/qsv_libav.h b/libhb/handbrake/qsv_libav.h
index b74b50c2be1757ecbbfc5fa67975b70a6bd20281..75b32db90329709de8919c67ee0a2877ce9946ce 100644
--- a/libhb/handbrake/qsv_libav.h
+++ b/libhb/handbrake/qsv_libav.h
@@ -26,8 +26,8 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#ifndef HB_QSV_LIBAV_H
-#define HB_QSV_LIBAV_H
+#ifndef HANDBRAKE_QSV_LIBAV_H
+#define HANDBRAKE_QSV_LIBAV_H
 
 /**
  * @file
@@ -478,4 +478,4 @@ void *hb_qsv_list_item(hb_qsv_list *, int);
 
 /* @} */
 
-#endif // HB_QSV_LIBAV_H
+#endif // HANDBRAKE_QSV_LIBAV_H
diff --git a/libhb/handbrake/qsv_memory.h b/libhb/handbrake/qsv_memory.h
index 3434114e17e80658016799ab6a37a15015c4147f..4c55d90a24ef8940a7657a0df2275f4b415191bb 100644
--- a/libhb/handbrake/qsv_memory.h
+++ b/libhb/handbrake/qsv_memory.h
@@ -26,8 +26,8 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 \* ********************************************************************* */
 
-#ifndef QSV_MEMORY_H
-#define QSV_MEMORY_H
+#ifndef HANDBRAKE_QSV_MEMORY_H
+#define HANDBRAKE_QSV_MEMORY_H
 
 #include "handbrake/project.h"
 
@@ -57,4 +57,4 @@ int qsv_nv12_to_yuv420(struct SwsContext* sws_context,hb_buffer_t* dst, mfxFrame
 int qsv_yuv420_to_nv12(struct SwsContext* sws_context,mfxFrameSurface1* dst, hb_buffer_t* src);
 
 #endif // HB_PROJECT_FEATURE_QSV
-#endif // QSV_MEMORY_H
+#endif // HANDBRAKE_QSV_MEMORY_H
diff --git a/libhb/handbrake/ssautil.h b/libhb/handbrake/ssautil.h
index 319262b736a194102b429780fbfcd1a2a15015ba..072c96e96148429d9a8a32c26ec92954d7699a1f 100644
--- a/libhb/handbrake/ssautil.h
+++ b/libhb/handbrake/ssautil.h
@@ -7,8 +7,8 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef __SSAUTIL_H__
-#define __SSAUTIL_H__
+#ifndef HANDBRAKE_SSAUTIL_H
+#define HANDBRAKE_SSAUTIL_H
 
 typedef struct hb_subtitle_style_s hb_subtitle_style_t;
 typedef struct hb_subtitle_style_context_s hb_subtitle_style_context_t;
@@ -31,4 +31,4 @@ void hb_muxmp4_process_subtitle_style(
         uint8_t  * input, uint8_t  ** output,
         uint8_t ** style, uint16_t  * stylesize);
 
-#endif // __SSAUTIL_H__
+#endif // HANDBRAKE_SSAUTIL_H
diff --git a/libhb/handbrake/taskset.h b/libhb/handbrake/taskset.h
index 35c21e1c000be273142d8dbcdd60a26f8142218e..9276cb38d7abac8385b0a07d01c7a0bc0381b472 100644
--- a/libhb/handbrake/taskset.h
+++ b/libhb/handbrake/taskset.h
@@ -7,8 +7,8 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#ifndef HB_TASKSET_H
-#define HB_TASKSET_H
+#ifndef HANDBRAKE_TASKSET_H
+#define HANDBRAKE_TASKSET_H
 
 #define TASKSET_POSIX_COMPLIANT 1
 
@@ -52,4 +52,4 @@ taskset_thread_stop( taskset_t *ts, int thr_idx )
     return bit_is_set( ts->task_stop_bitmap, thr_idx );
 }
 
-#endif /* HB_TASKSET_H */
+#endif /* HANDBRAKE_TASKSET_H */
diff --git a/libhb/handbrake/vce_common.h b/libhb/handbrake/vce_common.h
index 0c5da62b288b8caee50c5ad7ef076cf3c253cacb..3fa8a31df4ad6af6de97b3aff8bcf11d9b875c6c 100644
--- a/libhb/handbrake/vce_common.h
+++ b/libhb/handbrake/vce_common.h
@@ -7,6 +7,9 @@
  * For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
+#ifndef HANDBRAKE_VCE_COMMON_H
+#define HANDBRAKE_VCE_COMMON_H
+
 int            hb_vce_h264_available();
 int            hb_vce_h265_available();
 
@@ -18,3 +21,5 @@ static const char * const hb_vce_h264_level_names[] =
     "auto", "1.0", "1.1", "1.2", "1.3", "2.0", "2.1", "2.2", "3.0",
     "3.1", "3.2", "4.0", "4.1", "4.2", "5.0", "5.1", "5.2",  NULL,
 };
+
+#endif // HANDBRAKE_VCE_COMMON_H
