diff --git a/configure.ac b/configure.ac
index c004bd70d1b29be2151078af3ac0997132f3ec5a..b44ed1595af37c755569cbae3455298189fc5fb6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -826,7 +826,7 @@ AS_IF([test "x$enable_alsa" = "xyes" && test "x$HAVE_ALSA" = "x0"],
 AS_IF([test "x$HAVE_ALSA" = "x1"],
     [
         save_CPPFLAGS="$CPPFLAGS"; CPPFLAGS="$CPPFLAGS $ASOUNDLIB_CFLAGS"
-        AC_CHECK_HEADERS([use-case.h], HAVE_ALSA_UCM=1, HAVE_ALSA_UCM=0)
+        AC_CHECK_HEADERS([alsa/use-case.h], HAVE_ALSA_UCM=1, HAVE_ALSA_UCM=0)
         CPPFLAGS="$save_CPPFLAGS"
     ],
     HAVE_ALSA_UCM=0)
diff --git a/src/modules/alsa/alsa-ucm.h b/src/modules/alsa/alsa-ucm.h
index 53abf3f9084535732a889fa47f7c46adade55548..c926f3cc39a6ecaea56062f9e45acb41bd0c5965 100644
--- a/src/modules/alsa/alsa-ucm.h
+++ b/src/modules/alsa/alsa-ucm.h
@@ -23,7 +23,7 @@
 ***/
 
 #ifdef HAVE_ALSA_UCM
-#include <use-case.h>
+#include <alsa/use-case.h>
 #else
 typedef void snd_use_case_mgr_t;
 #endif
